diff -ur package_a/base-files/files/etc/openwrt_release package/base-files/files/etc/openwrt_release
--- a/package/base-files/files/etc/openwrt_release	2017-03-06 03:50:57.000000000 +0100
+++ b/package/base-files/files/etc/openwrt_release	2017-10-30 12:14:34.639257255 +0100
@@ -1,6 +1,6 @@
 DISTRIB_ID='%D'
 DISTRIB_RELEASE='%V'
-DISTRIB_REVISION='%R'
+DISTRIB_REVISION='%C'
 DISTRIB_CODENAME='%n'
 DISTRIB_TARGET='%S'
 DISTRIB_ARCH='%A'
