diff --git a/target/linux/mpc85xx/base-files/etc/uci-defaults/02_network b/target/linux/mpc85xx/base-files/etc/uci-defaults/02_network
index 10a2cc8..e4e3566 100755
--- a/target/linux/mpc85xx/base-files/etc/uci-defaults/02_network
+++ b/target/linux/mpc85xx/base-files/etc/uci-defaults/02_network
@@ -1,22 +1,5 @@
 #!/bin/sh
-#
-# Copyright (C) 2013 OpenWrt.org
-#
-
-tplink_set_mac()
-{
-	local cfg=$1
-	local offset=$2
-	local mac
-
-	. /lib/functions.sh
-
-	mac=$(mtd_get_mac_binary u-boot 326656)
-	mac=$(macaddr_add $mac $offset)
-
-	ucidef_set_interface_macaddr $cfg $mac
-}
-
+# Copyright (C) 2014 OpenWrt.org
 
 [ -e /etc/config/network ] && exit 0
 
@@ -24,6 +7,8 @@ touch /etc/config/network
 
 . /lib/functions/uci-defaults.sh
 . /lib/mpc85xx.sh
+. /lib/functions.sh
+. /lib/functions/system.sh
 
 ucidef_set_interface_loopback
 
@@ -35,10 +20,9 @@ tl-wdr4900-v1)
 	ucidef_add_switch "switch0" "1" "1"
 	ucidef_add_switch_vlan "switch0" "1" "0t 2 3 4 5"
 	ucidef_add_switch_vlan "switch0" "2" "0t 1"
-	tplink_set_mac lan -2
-	tplink_set_mac wan 1
+	ucidef_set_interface_macaddr lan "$(mtd_get_mac_binary config 338)"
+	ucidef_set_interface_macaddr wan "$(mtd_get_mac_binary config 344)"
 	;;
-
 *)
 	ucidef_set_interfaces_lan_wan "eth0" "eth1"
 	;;
