--- a/feeds/luci/contrib/package/community-profiles/files/etc/config/profile_berlin
+++ b/feeds/luci/contrib/package/community-profiles/files/etc/config/profile_berlin
@@ -1,7 +1,7 @@
 config 'community' 'profile'
 	option 'name' 'Freifunk Berlin'
 	option 'homepage' 'http://berlin.freifunk.net'
-	option 'ssid' 'freifunk.net'
+	option 'ssid' 'berlin.freifunk.net'
 	option 'ssid_scheme' 'addchannelbefore'
 	option 'mesh_network' '104.0.0.0/8'
 	option 'splash_network' '10.104.0.0/16'
