diff -ur openwrt.orig/feeds/luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js openwrt/feeds/luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js
--- openwrt.orig/feeds/luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js	2019-11-14 01:08:59.557322662 +0100
+++ openwrt/feeds/luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js	2019-11-14 01:10:22.742834697 +0100
@@ -769,7 +769,7 @@
 					o.wifiNetwork = radioNet;
 
 					o = ss.taboption('advanced', form.Flag, 'legacy_rates', _('Allow legacy 802.11b rates'));
-					o.default = o.enabled;
+					o.default = o.disabled;
 
 					o = ss.taboption('advanced', form.Value, 'distance', _('Distance Optimization'), _('Distance to farthest network member in meters.'));
 					o.datatype = 'range(0,114750)';
Nur in openwrt/feeds/luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network: wireless.js.orig.
