Index: openwrt/target/imagebuilder/files/Makefile
===================================================================
--- openwrt.orig/target/imagebuilder/files/Makefile
+++ openwrt/target/imagebuilder/files/Makefile
@@ -114,6 +114,9 @@ ifneq ($(USER_FILES),)
 	$(MAKE) copy_files
 endif
 	$(MAKE) package_postinst
+ifneq ($(CUSTOM_POSTINST_SCRIPT),)
+	(cd "$(TARGET_DIR)" ; bash $(CUSTOM_POSTINST_SCRIPT))
+endif
 	$(MAKE) build_image
 
 package_index: FORCE
