Index: openwrt/target/imagebuilder/files/Makefile
===================================================================
--- openwrt.orig/target/imagebuilder/files/Makefile
+++ openwrt/target/imagebuilder/files/Makefile
@@ -118,6 +118,9 @@ ifneq ($(USER_FILES),)
 	$(MAKE) copy_files
 endif
 	$(MAKE) -s package_postinst
+ifneq ($(CUSTOM_POSTINST_SCRIPT),)
+	(cd "$(TARGET_DIR)" ; bash $(CUSTOM_POSTINST_SCRIPT))
+endif
 	$(MAKE) -s build_image
 	$(MAKE) -s checksum
 
