Index: openwrt/package/base-files/files/etc/banner
===================================================================
--- openwrt.orig/package/base-files/files/etc/banner
+++ openwrt/package/base-files/files/etc/banner
@@ -1,14 +1,21 @@
-  _______                     ________        __
- |       |.-----.-----.-----.|  |  |  |.----.|  |_
- |   -   ||  _  |  -__|     ||  |  |  ||   _||   _|
- |_______||   __|_____|__|__||________||__|  |____|
-          |__| W I R E L E S S   F R E E D O M
+  _____        _  __             _
+ |  ___|      (_)/ _|           | |
+ | |_ _ __ ___ _| |_ _   _ _ __ | | __
+ |  _| '__/ _ \ |  _| | | | '_ \| |/ /
+ | | | | |  __/ | | | |_| | | | |   <
+ \_| |_|  \___|_|_|  \__,_|_| |_|_|\_\
+
  -----------------------------------------------------
- CHAOS CALMER (%C, %R)
- -----------------------------------------------------
-  * 1 1/2 oz Gin            Shake with a glassful
-  * 1/4 oz Triple Sec       of broken ice and pour
-  * 3/4 oz Lime Juice       unstrained into a goblet.
-  * 1 1/2 oz Orange Juice
-  * 1 tsp. Grenadine Syrup
+ Firmware Berlin (%N %C rev %R)
+ %P - %S
+ https://wiki.freifunk.net/Berlin:Firmware
+ https://github.com/freifunk-berlin
  -----------------------------------------------------
+
+ If you find bugs please report them at:
+
+   https://github.com/freifunk-berlin/firmware/issues
+
+ For questions write a mail to <berlin@berlin.freifunk.net>
+ or come to our weekly meetings at cbase/wikimedia in Berlin.
+
