Index: lede/package/base-files/files/etc/banner
===================================================================
--- lede.orig/package/base-files/files/etc/banner	2016-06-02 02:43:57.601381806 +0200
+++ lede/package/base-files/files/etc/banner	2016-06-02 02:46:17.928201514 +0200
@@ -1,10 +1,20 @@
-     _________
-    /        /\      _    ___ ___  ___
-   /  LE    /  \    | |  | __|   \| __|
-  /    DE  /    \   | |__| _|| |) | _|
- /________/  LE  \  |____|___|___/|___|                      lede-project.org
- \        \   DE /
-  \    LE  \    /  -----------------------------------------------------------
-   \  DE    \  /    %N (%V, %C)
-    \________\/    -----------------------------------------------------------
+  _____        _  __             _
+ |  ___|      (_)/ _|           | |
+ | |_ _ __ ___ _| |_ _   _ _ __ | | __
+ |  _| '__/ _ \ |  _| | | | '_ \| |/ /
+ | | | | |  __/ | | | |_| | | | |   <
+ \_| |_|  \___|_|_|  \__,_|_| |_|_|\_\
+
+ Firmware Berlin (%N %V rev %C)
+ %P - %S
+ https://wiki.freifunk.net/Berlin:Firmware
+ https://github.com/freifunk-berlin
+ -----------------------------------------------------
+
+ If you find bugs please report them at:
+
+   https://github.com/freifunk-berlin/firmware/issues
+
+ For questions write a mail to <berlin@berlin.freifunk.net>
+ or check https://berlin.freifunk.net/contact for our weekly meetings.
 
