Index: target/linux/atheros/files/arch/mips/atheros/ar5312/board.c
===================================================================
--- target/linux/atheros/files/arch/mips/atheros/ar5312/board.c	(revision 16280)
+++ target/linux/atheros/files/arch/mips/atheros/ar5312/board.c	(working copy)
@@ -151,7 +151,7 @@
 
 static struct resource ar5312_flash_resource = {
 	.start	= AR531X_FLASH,
-	.end	= AR531X_FLASH + 0x800000 - 1,
+	.end	= AR531X_FLASH + 0x400000 - 1,
 	.flags	= IORESOURCE_MEM,
 };
 
