Index: package/base-files/files/etc/config/system
===================================================================
--- package/base-files/files/etc/config/system	(Revision 21854)
+++ package/base-files/files/etc/config/system	(Arbeitskopie)
@@ -1,6 +1,7 @@
 config system
 	option hostname	OpenWrt
-	option timezone	UTC
+	option 'zonename' 'Europe/Berlin'
+	option 'timezone' 'CET-1CEST,M3.5.0,M10.5.0/3'
 
 config rdate
 	list server ac-ntp0.net.cmu.edu
