Index: package/base-files/files/lib/network/config.sh
===================================================================
--- package/base-files/files/lib/network/config.sh	(Revision 26751)
+++ package/base-files/files/lib/network/config.sh	(Arbeitskopie)
@@ -185,6 +185,7 @@
 					local stp
 					config_get_bool stp "$config" stp 0
 					$DEBUG brctl addbr "br-$config"
+					$DEBUG do_sysctl "net.ipv6.conf.br-$config.autoconf" 0
 					$DEBUG brctl setfd "br-$config" 0
 					$DEBUG ifconfig "$iface" 0.0.0.0
 					$DEBUG do_sysctl "net.ipv6.conf.$iface.disable_ipv6" 1
@@ -221,6 +222,8 @@
 	local iface="$1"
 	local config="$2"
 
+	$DEBUG do_sysctl "net.ipv6.conf.$iface.autoconf" 0
+
 	local ipaddr netmask ip6addr
 	config_get ipaddr "$config" ipaddr
 	config_get netmask "$config" netmask
