diff --git a/package/network/utils/comgt/Makefile b/package/network/utils/comgt/Makefile
index 5324903..9061675 100644
--- a/package/network/utils/comgt/Makefile
+++ b/package/network/utils/comgt/Makefile
@@ -25,7 +25,7 @@ define Package/comgt
   SECTION:=utils
   CATEGORY:=Utilities
   TITLE:=Option/Vodafone 3G/GPRS control tool
-  DEPENDS:=+chat
+  DEPENDS:=+chat +ppp
   URL:=http://www.pharscape.org/comgt.html
 endef
 
