Index: a/target/linux/atheros/image/Makefile
===================================================================
--- a/target/linux/atheros/image/Makefile	(Revision 33210)
+++ b/target/linux/atheros/image/Makefile	(Arbeitskopie)
@@ -56,6 +56,10 @@
 		"$(BIN_DIR)/$(IMG_PREFIX)-vmlinux.lzma" \
 		"$(BIN_DIR)/$(IMG_PREFIX)-root.$(1)" \
 		"$(BIN_DIR)/$(IMG_PREFIX)-combined.$(1).img"
+	-sh $(TOPDIR)/scripts/combined-image.sh \
+		"$(BIN_DIR)/$(IMG_PREFIX)-vmlinux.gz" \
+		"$(BIN_DIR)/$(IMG_PREFIX)-root.$(1)" \
+		"$(BIN_DIR)/$(IMG_PREFIX)-whr-hp-ag108.$(1).img"
   endif
 endef
 
