--- a/modules/freifunk/root/etc/config/freifunk	2010-08-25 19:46:41.135000064 +0200
+++ b/modules/freifunk/root/etc/config/freifunk.copy	2010-08-25 19:47:48.746000068 +0200
@@ -235,3 +235,14 @@
 
 
 
+config 'community' 'bno'
+	option 'name' 'Berlin NordOst'
+	option 'homepage' 'http://www.freifunk-bno.de/'
+	option 'ssid' 'olsr.freifunk.net'
+	option 'mesh_network' ' 104.0.0.0/8'
+	option 'splash_network' '10.104.0.0/16'
+	option 'splash_prefix' '27'
+	option 'external' 'freifunk_berlin'
+	option 'latitude' '52.55010'
+	option 'longitude' '13.45889'
+
+
+
