--- modules/freifunk/root/etc/config/freifunk	2010-08-25 19:46:41.135000064 +0200
+++ modules/freifunk/root/etc/config/freifunk.copy	2010-08-25 19:47:48.746000068 +0200
@@ -235,3 +235,14 @@
 
 
 
+config 'community' 'neuss'
+	option 'name' 'Freifunk Neuss'
+	option 'homepage' 'http://neuss.freifunk.net'
+	option 'ssid' 'neuss.freifunk.net'
+	option 'mesh_network' '172.28.0.0/16'
+	option 'splash_network' '10.104.0.0/16'
+	option 'splash_prefix' '27'
+	option 'external' 'freifunk_neuss'
+
+
+
--- modules/freifunk/root/etc/config/freifunk_neuss    1970-01-01 01:00:00.000000000 +0100
+++ modules/freifunk/root/etc/config/freifunk_neuss 2010-08-25 17:58:05.187000066 +0200
@@ -0,0 +1,10 @@
+package 'freifunk_neuss'
+
+config 'defaults' 'wifi_device'
+        option 'channel' '11'
+
+config 'defaults' 'interface'
+        option 'netmask' '255.255.0.0'
+
+config 'defaults' 'wifi_iface'
+        option 'bssid' 'DE:AD:BE:EF:CA:FE'
