--- modules/freifunk/root/etc/config/freifunk	2010-08-25 19:42:07.350000071 +0200
+++ modules/freifunk/root/etc/config/freifunk.copy	2010-08-25 19:43:35.657000069 +0200
@@ -224,3 +224,14 @@
 
 
 
+config 'community' 'pberg'
+	option 'name' 'Freifunk Berlin Prenzlauer Berg'
+	option 'homepage' 'http://pberg.freifunk.net'
+	option 'ssid' 'olsr.freifunk.net'
+	option 'mesh_network' '104.0.0.0/8'
+	option 'splash_network' '10.104.0.0/16'
+	option 'splash_prefix' '27'
+	option 'external' 'freifunk_berlin'
+
+
+
