--- a/modules/freifunk/root/etc/config/freifunk	2010-08-25 19:46:41.135000064 +0200
+++ b/modules/freifunk/root/etc/config/freifunk.copy	2010-08-25 19:47:48.746000068 +0200
@@ -235,3 +235,14 @@
 
 
 
+config 'community' 'nbd'
+	option 'name' 'Piraten Niederbayern'
+	option 'homepage' 'http://freifunk.piraten-niederbayern.de'
+	option 'ssid' 'Pirate-NDB.freifunk.net'
+	option 'mesh_network' ' 10.6.0.0/16'
+	option 'splash_network' '192.168.99.0/24'
+	option 'splash_prefix' '27'
+	option 'external' 'freifunk_ndb'
+	option 'latitude' '48.57325'
+	option 'longitude' '13.45074'
+
+
+
--- a/modules/freifunk/root/etc/config/freifunk_ndb    1970-01-01 01:00:00.000000000 +0100
+++ b/modules/freifunk/root/etc/config/freifunk_ndb 2010-08-25 17:58:05.187000066 +0200
@@ -0,0 +1,10 @@
+package 'freifunk_muenster'
+
+config 'defaults' 'wifi_device'
+        option 'channel' '10'
+
+config 'defaults' 'interface'
+        option 'netmask' '255.255.0.0'
+
+config 'defaults' 'wifi_iface'
+        option 'bssid' '02:CA:FF:EE:BA:BE'
