Index: target/imagebuilder/files/Makefile
===================================================================
--- target/imagebuilder/files/Makefile	(revision 18116)
+++ target/imagebuilder/files/Makefile	(working copy)
@@ -88,11 +88,6 @@
 BUILD_PACKAGES:=$(filter-out $(filter -%,$(BUILD_PACKAGES)) $(patsubst -%,%,$(filter -%,$(BUILD_PACKAGES))),$(BUILD_PACKAGES))
 
 image:
-	if [ -z "$($(PROFILE)_NAME)" ]; then \
-		echo Profile $(PROFILE) not found.; \
-		echo 'Use "make info" to get a list of available target profiles'; \
-		false; \
-	fi
 	echo 'Building images for $(BOARD) - $($(PROFILE)_NAME)'
 	echo 'Packages: $(BUILD_PACKAGES)'
 	echo
