Index: package/swconfig/files/switch.sh
===================================================================
--- package/swconfig/files/switch.sh	(revision 19108)
+++ package/swconfig/files/switch.sh	(working copy)
@@ -2,6 +2,9 @@
 # Copyright (C) 2009 OpenWrt.org
 
 setup_switch_dev() {
+	sysctl -w "net.ipv6.conf.$1.autoconf=0" > /dev/null 2> /dev/null
+	ifconfig "$1" 0.0.0.0 down
+	ifconfig "$1" up
 	swconfig dev "$1" load network
 }
 
Index: package/base-files/files/lib/network/config.sh
===================================================================
--- package/base-files/files/lib/network/config.sh	(revision 19108)
+++ package/base-files/files/lib/network/config.sh	(working copy)
@@ -160,6 +160,9 @@
 setup_interface_static() {
 	local iface="$1"
 	local config="$2"
+	
+        # Disable IPv6 autoconfiguration
+	sysctl -w "net.ipv6.conf.$iface.autoconf=0" > /dev/null 2> /dev/null
 
 	config_get ipaddr "$config" ipaddr
 	config_get netmask "$config" netmask

Index: package/base-files/files/etc/init.d/boot
===================================================================
--- package/base-files/files/etc/init.d/boot	(revision 19108)
+++ package/base-files/files/etc/init.d/boot	(working copy)
@@ -59,6 +59,9 @@
 		/usr/bin/env -i ACTION=add INTERFACE="$iface" /sbin/hotplug-call net
 	done
 
+	# Disable IPv6 autoconfiguration by default
+	sysctl -w "net.ipv6.conf.all.autoconf=0" > /dev/null 2> /dev/null
+
 	# create /dev/root if it doesn't exist
 	[ -e /dev/root ] || {
 		rootdev=$(awk 'BEGIN { RS=" "; FS="="; } $1 == "root" { print $2 }' < /proc/cmdline)
