diff --git a/contrib/package/remote-update/files/usr/sbin/remote-update b/contrib/package/remote-update/files/usr/sbin/remote-update
index 6e45e48..85cc427 100755
--- a/contrib/package/remote-update/files/usr/sbin/remote-update
+++ b/contrib/package/remote-update/files/usr/sbin/remote-update
@@ -6,13 +6,12 @@ local D4='\([0-9]\{4\}\)'
 local NL='
 '
 
-find_architecture()
+find_board()
 {
-	local ifs="$IFS"; IFS="-"
-	set -- $(opkg list_installed kernel)
+	local ifs="$IFS"; IFS=" "
+	set -- $(opkg list_installed base-files-*)
 	IFS="$ifs"
-
-	echo "$3"
+	echo $1 | sed -e 's/base-files-//'
 }
 
 find_image()
@@ -24,7 +23,7 @@ find_image()
 		ar71xx)
 			echo "openwrt-ar71xx-combined.squashfs.img"
 		;;
-		brcm)
+		brcm-2.4)
 			echo "openwrt-brcm-2.4-squashfs.trx"
 		;;
 	esac
@@ -168,13 +167,14 @@ local image_url="$updateurl"
 local image_name="${image_url##*/}"
 
 [ -z "$image_url" ] && {
-	local arch=$(find_architecture)
-	local image=$(find_image "$arch")
+	local board=$(find_board)
+	local image=$(find_image "$board")
 	local repo=$(uci get freifunk.upgrade.repository 2>/dev/null)
-	repo=${repo:-$(uci get system.upgrade.repository 2>/dev/null)}
+	repo=${repo:-$(uci get system.@upgrade[0].repository 2>/dev/null)}
+	local hostname=$(uci get system.@system[0].hostname 2>/dev/null)
 
-	[ -z "$arch" ] && {
-		echo "Can not determine the current architecture."
+	[ -z "$board" ] && {
+		echo "Can not determine the current board."
 		exit 1
 	}
 
@@ -185,16 +185,25 @@ local image_name="${image_url##*/}"
 	}
 
 	[ -z "$image" ] && {
-		echo "No suitable image for the '$arch' architecture."
+		echo "No suitable image for the '$board' board."
 		echo "Your platform is not supported."
 		exit 1
 	}
 
-	echo "Architecture: $arch"
+	echo "Board: $board"
 	echo "Repository:   $repo"
 
 	image_name="$image"
-	image_url="${repo%/}/$arch/$image"
+        if wget -q --spider "${repo%/}/$hostname/$board/$image" 2>/dev/null ; then
+	    image_url="${repo%/}/$hostname/$board/$image"
+        elif wget -q --spider "${repo%/}/$board/$image" 2>/dev/null ; then
+	    image_url="${repo%/}/$board/$image"
+        else
+		echo "No suitable image found at"
+		echo "${repo%/}/$board/$image"
+		exit 1
+	fi
+	    
 }
 
 
