Index: target/linux/atheros/Makefile
===================================================================
--- target/linux/atheros/Makefile	(revision 16280)
+++ target/linux/atheros/Makefile	(working copy)
@@ -22,3 +22,21 @@
 endef
 
 $(eval $(call BuildTarget))
+
+BOARDNAME:=Generic
+
+define Target/Description
+	Build firmware images for atheros based boards 
+	(e.g : fonera, dir300, ...)
+endef
+
+define Profile/Generic
+  NAME:=Generic
+  PACKAGES:= hostapd-mini kmod-madwifi
+endef
+
+define Profile/Generic/Description
+	Generic Profile
+endef
+
+$(eval $(call Profile,Generic))
