Index: package/dnsmasq/files/dnsmasq.init
===================================================================
--- package/dnsmasq/files/dnsmasq.init	(revision 18358)
+++ package/dnsmasq/files/dnsmasq.init	(working copy)
@@ -44,6 +44,10 @@
 	append args "-i $1"
 }
 
+append_addnhosts() {
+	append args "-H $1"
+}
+
 dnsmasq() {
 	local cfg="$1"
 	append_bool "$cfg" authoritative "-K"
@@ -68,6 +72,7 @@
 	append_parm "$cfg" ednspacket_max "-P"
 	append_parm "$cfg" dhcpleasemax "-X"
 	append_parm "$cfg" "addnhosts" "-H"
+	config_list_foreach "$cfg" append_addnhosts
 	append_parm "$cfg" "queryport" "-Q"
 	append_parm "$cfg" "domain" "-s"
 	append_parm "$cfg" "local" "-S"
