diff --git a/applications/luci-splash/root/etc/init.d/luci_splash b/applications/luci-splash/root/etc/init.d/luci_splash
index 351b9b3..31d2471 100755
--- a/applications/luci-splash/root/etc/init.d/luci_splash
+++ b/applications/luci-splash/root/etc/init.d/luci_splash
@@ -231,7 +231,7 @@ start() {
 	### Add crontab entry
 	test -f /etc/crontabs/root || touch /etc/crontabs/root
 	grep -q luci-splash /etc/crontabs/root || {
-		echo '*/5 * * * * 	/usr/sbin/luci-splash sync' >> /etc/crontabs/root
+		echo '0 */1 * * * 	/usr/sbin/luci-splash sync' >> /etc/crontabs/root
 	}
 
 	lock -u $LOCK
