diff --git a/modules/admin-mini/luasrc/controller/mini/system.lua b/modules/admin-mini/luasrc/controller/mini/system.lua
index 2559ac1..47bb1d4 100644
--- a/modules/admin-mini/luasrc/controller/mini/system.lua
+++ b/modules/admin-mini/luasrc/controller/mini/system.lua
@@ -152,16 +152,15 @@ function action_upgrade()
 			luci.http.prepare_content("text/plain")
 			luci.http.write("Starting luci-flash...\n")
 
+			io.flush()
+
 			-- Now invoke sysupgrade
 			local keepcfg = keep_avail and luci.http.formvalue("keepcfg") == "1"
-			local flash = ltn12_popen("/sbin/luci-flash %s %q" %{
-				keepcfg and "-k %q" % _keep_pattern() or "", tmpfile
+			local flash = ltn12_popen("/sbin/sysupgrade %s %q" %{
+				keepcfg and "" or "-n", tmpfile
 			})
 
 			luci.ltn12.pump.all(flash, luci.http.write)
-
-			-- Make sure the device is rebooted
-			luci.sys.reboot()
 		end
 
 
