diff --git a/modules/freifunk/root/etc/config/freifunk b/modules/freifunk/root/etc/config/freifunk
index fe1514f..8d5b7c7 100644
--- a/modules/freifunk/root/etc/config/freifunk
+++ b/modules/freifunk/root/etc/config/freifunk
@@ -89,6 +89,32 @@ config 'defaults' 'upgrade'
 	option 'repository' 'http://dev.luci.freifunk-halle.net/freifunk-snapshots'
 	option 'rssfeed' 'http://firmware.leipzig.freifunk.net/kamikaze/.rss.xml'
 
+config 'defaults' 'radvd_interface'
+	option 'AdvSendAdvert' '1'
+	option 'AdvManagedFlag' '0'
+	option 'AdvOtherConfigFlag' '0'
+	option 'MinRtrAdvInterval' '3'
+	option 'MaxRtrAdvInterval' '6'
+	option 'AdvLinkMTU' '1280'
+	option 'ignore' '0'
+
+config 'defaults' 'radvd_prefix'
+	option 'AdvOnLink' '1'
+	option 'AdvAutonomous' '1'
+	option 'AdvRouterAddr' '0'
+	option 'AdvPreferredLifetime' '6'
+	option 'AdvValidLifetime' '12'
+	option 'ignore' '0'
+
+config 'defaults' 'radvd_rdnss'
+	option 'AdvRDNSSLifetime' '60'
+	option 'ignore' '0'
+
+config 'defaults' 'radvd_dnssl'
+	option 'AdvDNSSLLifetime' '60'
+	option 'suffix' 'olsr'
+	option 'ignore' '0'
+
 config 'community' 'leipzig'
 	option 'name' 'Freifunk Leipzig'
 	option 'homepage' 'http://leipzig.freifunk.net'
