Index: modules/freifunk/luasrc/model/cbi/freifunk/public_status.lua
===================================================================
--- modules/freifunk/luasrc/model/cbi/freifunk/public_status.lua	(Revision 6252)
+++ modules/freifunk/luasrc/model/cbi/freifunk/public_status.lua	(Arbeitskopie)
@@ -208,9 +208,13 @@
 	metric = v6:option(DummyValue, "metric", translate("metric"))
 	function metric.cfgvalue(self, section)
 		local metr = routes6[section].metric
-		local lower = bit.band(metr, 0xffff)
-		local higher = bit.rshift(bit.band(metr, 0xffff0000), 16)
-		return "%04X%04X" % {higher, lower}
+		if type(metr) == "integer" then
+			local lower = bit.band(metr, 0xffff)
+			local higher = bit.rshift(bit.band(metr, 0xffff0000), 16)
+			return "%04X%04X" % {higher, lower}
+		else
+			return metr
+		end
 	end
 end
 
