diff --git a/contrib/package/freifunk-firewall/files/etc/firewall.freifunk b/contrib/package/freifunk-firewall/files/etc/firewall.freifunk
index 9052992..3b06716 100644
--- a/contrib/package/freifunk-firewall/files/etc/firewall.freifunk
+++ b/contrib/package/freifunk-firewall/files/etc/firewall.freifunk
@@ -37,21 +37,6 @@ apply_advanced() {
 config_foreach apply_advanced advanced
 
 
-#
-# Apply fixes for masquerading rules
-#
-apply_nat_fix() {
-	local up
-	local ifname
-	config_get up $1 up
-	[ -n "$up" ] || return 0
-	(ACTION="ifup" INTERFACE="$1" . /etc/hotplug.d/iface/22-firewall-nat-fix )
-}
-
-uci_set_state firewall core loaded 1
-config_foreach fw_addif interface
-config_foreach apply_nat_fix interface
-
 [ -x /etc/init.d/luci_splash ] && ( sleep 3; /etc/init.d/luci_splash restart )&
 
 [ -x /etc/init.d/freifunk-p2pblock ] && /etc/init.d/freifunk-p2pblock enabled && \
