diff --git a/contrib/package/community-profiles/files/etc/config/profile_berlin b/contrib/package/community-profiles/files/etc/config/profile_berlin
index 94d6cda..9428480 100644
--- a/contrib/package/community-profiles/files/etc/config/profile_berlin
+++ b/contrib/package/community-profiles/files/etc/config/profile_berlin
@@ -1,7 +1,7 @@
 config 'community' 'profile'
 	option 'name' 'Freifunk Berlin'
 	option 'homepage' 'http://berlin.freifunk.net'
-	option 'ssid' 'olsr.freifunk.net'
+	option 'ssid' 'ch13.freifunk.net'
 	option 'mesh_network' '104.0.0.0/8'
 	option 'splash_network' '10.104.0.0/16'
 	option 'splash_prefix' '27'
@@ -12,11 +12,16 @@ config 'community' 'profile'
         option 'mapserver' 'http://map.pberg.freifunk.net/'
 
 config 'defaults' 'wifi_device'
-	option 'channel' '10'
+	list 'channel' '13'
+	list 'channel' '40'
+	list 'channel' '100'
 
 config 'defaults' 'bssidscheme'
 	option '10'	'02:CA:FF:EE:BA:BE'
 
+config 'defaults' 'ssidscheme'
+	option '10' 'olsr.freifunk.net'
+
 config 'defaults' 'interface'
 	option 'netmask' '255.0.0.0'
 
diff --git a/contrib/package/community-profiles/files/etc/config/profile_bno b/contrib/package/community-profiles/files/etc/config/profile_bno
index 2d0f0f5..bc55266 100644
--- a/contrib/package/community-profiles/files/etc/config/profile_bno
+++ b/contrib/package/community-profiles/files/etc/config/profile_bno
@@ -1,7 +1,7 @@
 config 'community' 'profile'
 	option 'name' 'Berlin NordOst'
 	option 'homepage' 'http://www.freifunk-bno.de/'
-	option 'ssid' 'olsr.freifunk.net'
+	option 'ssid' 'ch13.freifunk.net'
 	option 'mesh_network' '104.0.0.0/8'
 	option 'splash_network' '10.104.0.0/16'
 	option 'splash_prefix' '27'
@@ -13,7 +13,15 @@ config 'community' 'profile'
         option 'mapserver' 'http://map.pberg.freifunk.net/'
 
 config 'defaults' 'wifi_device'
-	option 'channel' '10'
+	list 'channel' '13'
+	list 'channel' '40'
+	list 'channel' '100'
+
+config 'defaults' 'bssidscheme'
+	option '10' '02:CA:FF:EE:BA:BE'
+
+config 'defaults' 'ssidscheme'
+	option '10' 'olsr.freifunk.net'
 
 config 'defaults' 'interface'
 	option 'netmask' '255.0.0.0'
diff --git a/contrib/package/community-profiles/files/etc/config/profile_pberg b/contrib/package/community-profiles/files/etc/config/profile_pberg
index 1122eaa..ca45043 100644
--- a/contrib/package/community-profiles/files/etc/config/profile_pberg
+++ b/contrib/package/community-profiles/files/etc/config/profile_pberg
@@ -1,7 +1,7 @@
 config 'community' 'profile'
 	option 'name' 'Freifunk Berlin Prenzlauer Berg'
 	option 'homepage' 'http://pberg.freifunk.net'
-	option 'ssid' 'olsr.freifunk.net'
+	option 'ssid' 'ch13.freifunk.net'
 	option 'mesh_network' '104.0.0.0/8'
 	option 'splash_network' '10.104.0.0/16'
 	option 'splash_prefix' '27'
@@ -12,7 +12,15 @@ config 'community' 'profile'
 	option 'mapserver' 'http://map.pberg.freifunk.net/'
 
 config 'defaults' 'wifi_device'
-	option 'channel' '10'
+	list 'channel' '13'
+	list 'channel' '40'
+	list 'channel' '100'
+
+config 'defaults' 'bssidscheme'
+	option '10' '02:CA:FF:EE:BA:BE'
+
+config 'defaults' 'ssidscheme'
+	option '10' 'olsr.freifunk.net'
 
 config 'defaults' 'interface'
 	option 'netmask' '255.0.0.0'
