diff --git a/applications/luci-olsr/luasrc/controller/olsr.lua b/applications/luci-olsr/luasrc/controller/olsr.lua
index 44849df..c796dc2 100644
--- a/applications/luci-olsr/luasrc/controller/olsr.lua
+++ b/applications/luci-olsr/luasrc/controller/olsr.lua
@@ -139,7 +139,7 @@ function action_neigh(json)
 	local devices  = ntm:get_wifidevs()
 	local sys = require "luci.sys"
 	local assoclist = {}
-	local neightbl = require "neightbl"
+	--local neightbl = require "neightbl"
 	local ipc = require "luci.ip"
 
 	luci.sys.net.routes(function(r) 
@@ -209,7 +209,9 @@ function action_neigh(json)
 						if not a:is6linklocal() then
 							if a:host() == locip:host() then
 								interface = name
-								neihgt = neightbl.get(device.ifname) or {}
+								if neightbl then
+									neihgt = neightbl.get(device.ifname) or {}
+								end
 							end
 						end
 					end
