diff --git a/applications/luci-statistics/root/usr/bin/stat-genconfig b/applications/luci-statistics/root/usr/bin/stat-genconfig
index 4c396ff..5d51682 100755
--- a/applications/luci-statistics/root/usr/bin/stat-genconfig
+++ b/applications/luci-statistics/root/usr/bin/stat-genconfig
@@ -364,6 +364,11 @@ plugins = {
 		{ },
 		{ }
 	},
+	madwifi = {
+		{ "WatchSet" },
+		{ },
+		{ "Interfaces", "WatchAdds" }
+	},
 }
 
 preprocess = {
