diff --git a/contrib/package/freifunk-gwcheck/files/usr/sbin/ff_olsr_test_gw.sh b/contrib/package/freifunk-gwcheck/files/usr/sbin/ff_olsr_test_gw.sh
index 5d34d9a..8664964 100755
--- a/contrib/package/freifunk-gwcheck/files/usr/sbin/ff_olsr_test_gw.sh
+++ b/contrib/package/freifunk-gwcheck/files/usr/sbin/ff_olsr_test_gw.sh
@@ -3,7 +3,7 @@
 #check if dyngw_plain is installed and enabled, else exit
 dyngwplainlib=`uci show olsrd |grep dyn_gw_plain |awk {' FS="."; print $1"."$2 '}`
 if [ -n "$dyngwplainlib" ]; then
-	if [ ! "$(uci -q get $dyngwplainlib.ignore)" == 0 ]; then
+	if [ "$(uci -q get $dyngwplainlib.ignore)" == 1 ]; then
 		exit 1
 	fi
 else
@@ -15,8 +15,8 @@ fi
 # check if we have a defaultroute with metric=0 in one of these tables: main table and gw-check table.
 # If not exit here.
 defroutemain="$(ip r s |grep default |grep -v metric)"
-defroutegw-check="$(ip r s t gw-check |grep default |grep -v metric)"
-if [ -z "$defroutegw-check" -a -z "$defroutemain" ]; then
+defroutegwcheck="$(ip r s t gw-check |grep default |grep -v metric)"
+if [ -z "$defroutegwcheck" -a -z "$defroutemain" ]; then
 	exit 1
 fi
 
@@ -57,14 +57,15 @@ iw=$(check_internet)
 if [ "$iw" == 0 ]; then
 	# check if we have a seperate routing table for our tests.
 	# If yes, move defaultroute to normal table and delete table gw-check
-	if [ -n "$defroutegw-check" ]; then
-		ip r a $defroutegw-check
-		ip r d $defroutegw-check t gw-check
+	if [ -n "$defroutegwcheck" ]; then
+		ip r a $defroutegwcheck
+		ip r d $defroutegwcheck t gw-check
 		ip ru del fwmark 0x2 lookup gw-check
 		for host in $testserver; do
-			iptables -t mangle -D OUTPUT -d $host -p tcp --dport 80 -j MARK --set-mark 0x2
+			hostip=$(nslookup $host 2>/dev/null | tail -n1 | cut -d ' ' -f 3)
+			iptables -t mangle -D OUTPUT -d $hostip -p tcp --dport 80 -j MARK --set-mark 0x2
 		done
-		logger -t gw-check "Internet is available again, restoring default route ( $defroutegw-check)"
+		logger -t gw-check "Internet is available again, restoring default route ( $defroutegwcheck)"
 	fi
 	
 else
@@ -73,7 +74,8 @@ else
 	if [ -z "$(ip ru s | grep gw-check)" -a -n "$defroutemain" ]; then
 		ip rule add fwmark 0x2 lookup gw-check
 		for host in $testserver; do
-			iptables -t mangle -I OUTPUT -d $host -p tcp --dport 80 -j MARK --set-mark 0x2
+			hostip=$(nslookup $host 2>/dev/null | tail -n1 | cut -d ' ' -f 3)
+			iptables -t mangle -I OUTPUT -d $hostip -p tcp --dport 80 -j MARK --set-mark 0x2
 		done
 		ip r a $defroutemain table gw-check
 		ip r d $defroutemain
diff --git a/contrib/package/freifunk-gwcheck/root/etc/uci-defaults/freifunk-gwcheck b/contrib/package/freifunk-gwcheck/root/etc/uci-defaults/freifunk-gwcheck
index e04faa5..78bec36 100644
--- a/contrib/package/freifunk-gwcheck/root/etc/uci-defaults/freifunk-gwcheck
+++ b/contrib/package/freifunk-gwcheck/root/etc/uci-defaults/freifunk-gwcheck
@@ -1,5 +1,6 @@
 #!/bin/sh
 tables="/etc/iproute2/rt_tables"
+mkdir -p /etc/iproute2/
 grep -q "gw-check" $tables || echo "200 gw-check" >> $tables
 test -f /etc/crontabs/root || touch /etc/crontabs/root
 grep -q "ff_olsr_test_gw.sh" /etc/crontabs/root || echo "* * * * *	/usr/sbin/ff_olsr_test_gw.sh" >> /etc/crontabs/root
