diff --git a/modules/freifunk/root/etc/config/freifunk b/modules/freifunk/root/etc/config/freifunk
index 8d5b7c7..84ed632 100644
--- a/modules/freifunk/root/etc/config/freifunk
+++ b/modules/freifunk/root/etc/config/freifunk
@@ -115,6 +115,26 @@ config 'defaults' 'radvd_dnssl'
 	option 'suffix' 'olsr'
 	option 'ignore' '0'
 
+config 'defaults' 'olsr_gvpninterface'
+	option 'ignore' '0'
+	option 'interface' 'gvpn'
+	option 'HelloInterval' '10.0'
+	option 'HelloValidityTime' '900.0'
+	option 'MidInterval' '150.0'
+	option 'HnaInterval' '150.0'
+	option 'HnaValidityTime' '900.0'
+	option 'Ip4Broadcast' '255.255.255.255'
+	option 'MidValidityTime' '900.0'
+
+config 'defaults' 'gvpn_node'
+	option 'supernode' 'ns1.pberg.freifunk.net:8719'
+	option 'tundev' 'gvpn'
+	option 'community' 'bbb'
+	option 'localport' '8719'
+	option 'subnet' '255.255.255.192'
+	option 'up' '/etc/l2gvpn/up_down'
+	option 'down' '/etc/l2gvpn/up_down'
+
 config 'community' 'leipzig'
 	option 'name' 'Freifunk Leipzig'
 	option 'homepage' 'http://leipzig.freifunk.net'
