Index: applications/luci-olsr/luasrc/model/cbi/olsr/olsrdplugins.lua
===================================================================
--- applications/luci-olsr/luasrc/model/cbi/olsr/olsrdplugins.lua	(revision 6195)
+++ applications/luci-olsr/luasrc/model/cbi/olsr/olsrdplugins.lua	(working copy)
@@ -139,9 +139,14 @@
 		},
 
 		["olsrd_mdns.so.1.0.0"] = {
-			{ DynamicList,	"NonOlsrIf",			"br-lan" }
+			{ DynamicList,	"NonOlsrIf",			"lan" }
 		},
 
+		["olsrd_p2pd.so.0.1.0"] = {
+			{ DynamicList,	"NonOlsrIf",			"lan" },
+			{ Value,	"P2pdTtl",			"10" }
+		},
+
 		["olsrd_arprefresh.so.0.1"]		= {},
 		["olsrd_dot_draw.so.0.3"]		= {},
 		["olsrd_dyn_gw_plain.so.0.4"]	= {},
