diff --git a/applications/luci-splash/luasrc/controller/splash/splash.lua b/applications/luci-splash/luasrc/controller/splash/splash.lua
index 27eff65..d48de72 100644
--- a/applications/luci-splash/luasrc/controller/splash/splash.lua
+++ b/applications/luci-splash/luasrc/controller/splash/splash.lua
@@ -6,12 +6,15 @@ function index()
 
 	entry({"admin", "services", "splash"}, cbi("splash/splash"), luci.i18n.translate("Client-Splash"), 90)
 	entry({"admin", "services", "splash", "splashtext" }, form("splash/splashtext"), luci.i18n.translate("Splashtext"), 10)
+	entry({"mini", "services", "splash"}, cbi("splash/splash"), luci.i18n.translate("Client-Splash"), 90)
+	entry({"mini", "services", "splash", "splashtext" }, form("splash/splashtext"), luci.i18n.translate("Splashtext"), 10)
 
 	node("splash").target = call("action_dispatch")
 	node("splash", "activate").target = call("action_activate")
 	node("splash", "splash").target   = template("splash_splash/splash")
 
 	entry({"admin", "status", "splash"}, call("action_status_admin"), "Client-Splash")
+	entry({"mini", "services", "splash_status"}, call("action_status_admin"), "Client-Splash-Status")
 end
 
 function action_dispatch()
