diff --git a/modules/freifunk/root/etc/config/freifunk b/modules/freifunk/root/etc/config/freifunk
index 327957c..9283e07 100644
--- a/modules/freifunk/root/etc/config/freifunk
+++ b/modules/freifunk/root/etc/config/freifunk
@@ -82,6 +82,26 @@ config 'defaults' 'dhcp'
 config 'defaults' 'olsr_interface'
 	option 'Ip4Broadcast' '255.255.255.255'
 
+config 'defaults' 'olsr_gvpninterface'
+	option 'ignore' '0'
+	option 'interface' 'gvpn'
+	option 'HelloInterval' '10.0'
+	option 'HelloValidityTime' '900.0'
+	option 'MidInterval' '150.0'
+	option 'HnaInterval' '150.0'
+	option 'HnaValidityTime' '900.0'
+	option 'Ip4Broadcast' '255.255.255.255'
+	option 'MidValidityTime' '900.0'
+
+config 'defaults' 'gvpn_node'
+	option 'supernode' 'ns1.pberg.freifunk.net:8719'
+	option 'tundev' 'gvpn'
+	option 'community' 'bbb'
+	option 'localport' '8719'
+	option 'subnet' '255.255.255.192'
+	option 'up' '/etc/l2gvpn/up_down'
+	option 'down' '/etc/l2gvpn/up_down'
+
 config 'defaults' 'time'
 	option 'rdate_servers' '128.138.140.44 171.64.7.77 171.64.7.99 81.169.154.44 130.133.1.10'
 
@@ -349,3 +369,5 @@ config 'community' 'pberg'
 	option 'external' 'freifunk_berlin'
 
 
+
+
