diff --git a/contrib/package/freifunk-common/files/etc/config/freifunk b/contrib/package/freifunk-common/files/etc/config/freifunk
index 09d92d9..cc14cca 100644
--- a/contrib/package/freifunk-common/files/etc/config/freifunk
+++ b/contrib/package/freifunk-common/files/etc/config/freifunk
@@ -127,3 +127,49 @@ config 'defaults' 'upgrade'
 	option 'repository' 'http://dev.luci.freifunk-halle.net/freifunk-snapshots'
 	option 'rssfeed' 'http://firmware.leipzig.freifunk.net/kamikaze/.rss.xml'
 
+config 'defaults' 'radvd_interface'
+	option 'AdvSendAdvert' '1'
+	option 'AdvManagedFlag' '0'
+	option 'AdvOtherConfigFlag' '0'
+	option 'MinRtrAdvInterval' '3'
+	option 'MaxRtrAdvInterval' '6'
+	option 'AdvLinkMTU' '1280'
+	option 'ignore' '0'
+
+config 'defaults' 'radvd_prefix'
+	option 'AdvOnLink' '1'
+	option 'AdvAutonomous' '1'
+	option 'AdvRouterAddr' '0'
+	option 'AdvPreferredLifetime' '6'
+	option 'AdvValidLifetime' '12'
+	option 'ignore' '0'
+
+config 'defaults' 'radvd_rdnss'
+	option 'AdvRDNSSLifetime' '60'
+	option 'ignore' '0'
+
+config 'defaults' 'radvd_dnssl'
+	option 'AdvDNSSLLifetime' '60'
+	option 'suffix' 'olsr'
+	option 'ignore' '0'
+
+config 'defaults' 'olsr_gvpninterface'
+	option 'ignore' '0'
+	option 'interface' 'gvpn'
+	option 'HelloInterval' '10.0'
+	option 'HelloValidityTime' '900.0'
+	option 'MidInterval' '150.0'
+	option 'HnaInterval' '150.0'
+	option 'HnaValidityTime' '900.0'
+	option 'Ip4Broadcast' '255.255.255.255'
+	option 'MidValidityTime' '900.0'
+
+config 'defaults' 'gvpn_node'
+	option 'supernode' 'ns1.pberg.freifunk.net:8719'
+	option 'tundev' 'gvpn'
+	option 'community' 'bbb'
+	option 'localport' '8719'
+	option 'subnet' '255.255.255.192'
+	option 'up' '/etc/l2gvpn/up_down'
+	option 'down' '/etc/l2gvpn/up_down'
+
