diff --git a/applications/luci-splash/luasrc/controller/splash/splash.lua b/applications/luci-splash/luasrc/controller/splash/splash.lua
index aceeb4a..8e1a263 100644
--- a/applications/luci-splash/luasrc/controller/splash/splash.lua
+++ b/applications/luci-splash/luasrc/controller/splash/splash.lua
@@ -7,6 +7,8 @@ local util = require "luci.util"
 function index()
 	entry({"admin", "services", "splash"}, cbi("splash/splash"), _("Client-Splash"), 90).i18n = "freifunk"
 	entry({"admin", "services", "splash", "splashtext" }, form("splash/splashtext"), _("Splashtext"), 10)
+	entry({"mini", "system", "splash"}, cbi("splash/splash"), _("Client-Splash"), 90)
+	entry({"mini", "system", "splash", "splashtext" }, form("splash/splashtext"), _("Splashtext"), 10)
 
 	local e
 	
@@ -19,6 +21,7 @@ function index()
 	node("splash", "blocked").target  = template("splash/blocked")
 
 	entry({"admin", "status", "splash"}, call("action_status_admin"), _("Client-Splash"))
+	entry({"mini", "status", "splash"}, call("action_status_admin"), _("Client-Splash-Status"))
 
 	local page  = node("splash", "publicstatus")
 	page.target = call("action_status_public")
