--- /dev/null	2011-07-21 00:06:42.963766673 +0200
+++ b/contrib/package/community-profiles/files/etc/config/profile_ffwtal	2011-07-28 17:11:02.840420605 +0200
@@ -0,0 +1,38 @@
+config 'community' 'profile'
+	option 'name' 'Freifunk Wuppertal'
+	option 'homepage' 'http://wuppertal.freifunk.net'
+	option 'ssid' 'wuppertal.freifunk.net'
+	option 'mesh_network' '10.3.0.0/16'
+	option 'splash_network' '192.168.101.0/24'
+	option 'splash_prefix' '27'
+	option 'latitude' '51.25472'
+	option 'longitude' '7.15000'
+	option 'suffix' 'olsr'
+
+config 'defaults' 'wifi_device'
+	option 'channel' '11'
+
+config 'defaults' 'wifi_iface'
+        option 'bssid' '02:40:00:42:42:42'
+
+config 'defaults' 'interface'
+        option 'netmask' '255.255.0.0'
+
+config 'defaults' 'olsr_interface'
+	option 'Ip4Broadcast' '255.255.255.255'
+	option 'HelloInterval' '3.0'
+	option 'HelloValidityTime' '125.0'
+	option 'TcInterval'  '2.0'
+	option 'TcValidityTime' '500.0'
+	option 'MidInterval' '25.0'
+	option 'MidValidityTime'  '500.0'
+	option 'HnaInterval' '10.0'
+	option 'HnaValidityTime' '125.0'
+
+config 'defaults' 'olsrd'
+	option 'AllowNoInt' 'yes'
+	option 'FIBMetric' 'flat'
+	option 'Pollrate' '0.025'
+	option 'TcRedundancy' '2'
+	option 'NatThreshold' '0.75'
+	option 'LinkQualityAlgorithm' 'etx_ff'
