diff --git a/modules/freifunk/luasrc/controller/freifunk/freifunk.lua b/modules/freifunk/luasrc/controller/freifunk/freifunk.lua
index 666f086..68a0d1a 100644
--- a/modules/freifunk/luasrc/controller/freifunk/freifunk.lua
+++ b/modules/freifunk/luasrc/controller/freifunk/freifunk.lua
@@ -53,9 +53,7 @@ function index()
 	page.setuser  = false
 	page.setgroup = false
 
-	entry({"freifunk", "status.json"}, call("jsonstatus"))
 	entry({"freifunk", "status", "zeroes"}, call("zeroes"), "Testdownload")
-	entry({"freifunk", "status", "public_status_json"}, call("public_status_json")).leaf = true
 
 	if nixio.fs.access("/usr/sbin/luci-splash") then
 		assign({"freifunk", "status", "splash"}, {"splash", "publicstatus"}, _("Splash"), 40)
@@ -68,42 +66,49 @@ function index()
 	end
 
 	-- backend
-	assign({"mini", "freifunk"}, {"admin", "freifunk"}, _("Freifunk"), 5)
-	entry({"admin", "freifunk"}, alias("admin", "freifunk", "index"), _("Freifunk"), 5)
+	entry({"admin", "freifunk"}, alias("admin", "freifunk", "ffwizard"), _("Freifunk"), 5)
+	entry({"mini", "freifunk"}, alias("mini", "freifunk", "ffwizard"), _("Freifunk"), 5)
 
-	page        = node("admin", "freifunk")
+	page        = node("admin", "freifunk", "index")
 	page.target = template("freifunk/adminindex")
 	page.title  = _("Freifunk")
 	page.order  = 5
+	assign({"mini", "freifunk", "index"}, {"admin", "freifunk", "index"}, _("Freifunk"), 5)
 
 	page        = node("admin", "freifunk", "basics")
 	page.target = cbi("freifunk/basics")
 	page.title  = _("Basic Settings")
 	page.order  = 5
-	
+	assign({"mini", "freifunk", "basics"}, {"admin", "freifunk", "basics"}, _("Basic Settings"), 5)
+
 	page        = node("admin", "freifunk", "basics", "profile")
 	page.target = cbi("freifunk/profile")
 	page.title  = _("Profile")
 	page.order  = 10
+	assign({"mini", "freifunk", "basics", "profile"}, {"admin", "freifunk", "basics", "profile"}, _("Profile"), 10)
 
 	page        = node("admin", "freifunk", "basics", "profile_expert")
 	page.target = cbi("freifunk/profile_expert")
 	page.title  = _("Profile (Expert)")
 	page.order  = 20
+	assign({"mini", "freifunk", "basics", "profile_expert"}, {"admin", "freifunk", "basics", "profile_expert"}, _("Profile (Expert)"), 20)
 
 	page        = node("admin", "freifunk", "Index-Page")
 	page.target = cbi("freifunk/user_index")
 	page.title  = _("Index Page")
 	page.order  = 50
+	assign({"mini", "freifunk", "Index-Page"}, {"admin", "freifunk", "Index-Page"}, _("Index Page"), 50)
 
 	page        = node("admin", "freifunk", "contact")
 	page.target = cbi("freifunk/contact")
 	page.title  = _("Contact")
 	page.order  = 15
+	assign({"mini", "freifunk", "contact"}, {"admin", "freifunk", "contact"}, _("Contact"), 15)
 
 	entry({"freifunk", "map"}, template("freifunk-map/frame"), _("Map"), 50)
 	entry({"freifunk", "map", "content"}, template("freifunk-map/map"), nil, 51)
 	entry({"admin", "freifunk", "profile_error"}, template("freifunk/profile_error"))
+	assign({"mini", "freifunk", "profile_error"}, {"admin", "freifunk", "profile_error"})
 end
 
 local function fetch_olsrd()
