diff --git a/applications/luci-olsr/luasrc/model/cbi/olsr/olsrd.lua b/applications/luci-olsr/luasrc/model/cbi/olsr/olsrd.lua
index 9bdf3a3..dd9eac1 100644
--- a/applications/luci-olsr/luasrc/model/cbi/olsr/olsrd.lua
+++ b/applications/luci-olsr/luasrc/model/cbi/olsr/olsrd.lua
@@ -342,7 +342,7 @@ av.write = write_float
 ifs = m:section(TypedSection, "Interface", translate("Interfaces"))
 ifs.addremove = true
 ifs.anonymous = true
-ifs.extedit   = luci.dispatcher.build_url("admin/services/olsrd/iface/%s")
+ifs.extedit   = luci.dispatcher.build_url("mini/services/olsrd/iface/%s")
 ifs.template  = "cbi/tblsection"
 
 function ifs.create(...)
diff --git a/applications/luci-olsr/luasrc/model/cbi/olsr/olsrdiface.lua b/applications/luci-olsr/luasrc/model/cbi/olsr/olsrdiface.lua
index 25692fd..2a5c9ce 100644
--- a/applications/luci-olsr/luasrc/model/cbi/olsr/olsrdiface.lua
+++ b/applications/luci-olsr/luasrc/model/cbi/olsr/olsrdiface.lua
@@ -26,7 +26,7 @@ m = Map("olsrd", translate("OLSR Daemon - Interface"),
 	"It runs on any wifi card that supports ad-hoc mode and of course on any ethernet device. "..
 	"Visit <a href='http://www.olsr.org'>olsrd.org</a> for help and documentation."))
 
-m.redirect = luci.dispatcher.build_url("admin/services/olsrd")
+m.redirect = luci.dispatcher.build_url("mini/services/olsrd")
 
 if not arg[1] or m.uci:get("olsrd", arg[1]) ~= "Interface" then
 	luci.http.redirect(m.redirect)
diff --git a/applications/luci-olsr/luasrc/model/cbi/olsr/olsrdplugins.lua b/applications/luci-olsr/luasrc/model/cbi/olsr/olsrdplugins.lua
index bd72c56..8deadc5 100644
--- a/applications/luci-olsr/luasrc/model/cbi/olsr/olsrdplugins.lua
+++ b/applications/luci-olsr/luasrc/model/cbi/olsr/olsrdplugins.lua
@@ -253,7 +253,7 @@ else
 	t.override_scheme = true
 	function t.extedit(self, section)
 		local lib = self.map:get(section, "library") or ""
-		return luci.dispatcher.build_url("admin", "services", "olsrd", "plugins") .. "/" .. lib
+		return luci.dispatcher.build_url("mini", "services", "olsrd", "plugins") .. "/" .. lib
 	end
 
 	ign = t:option( Flag, "ignore", translate("Enabled") )
