Index: target/linux/ar71xx/files/arch/mips/ar71xx/mach-tl-wr741nd.c
===================================================================
--- target/linux/ar71xx/files/arch/mips/ar71xx/mach-tl-wr741nd.c	(Revision 26870)
+++ target/linux/ar71xx/files/arch/mips/ar71xx/mach-tl-wr741nd.c	(Arbeitskopie)
@@ -44,10 +44,9 @@
 		.offset		= 0x160000,
 		.size		= 0x290000,
 	} , {
-		.name		= "art",
+		.name		= "aarrtt",
 		.offset		= 0x3f0000,
 		.size		= 0x010000,
-		.mask_flags	= MTD_WRITEABLE,
 	} , {
 		.name		= "firmware",
 		.offset		= 0x020000,
