diff --git a/net/olsrd/files/olsrd.init b/net/olsrd/files/olsrd.init
index dd82bb5..0d4b5e3 100644
--- a/net/olsrd/files/olsrd.init
+++ b/net/olsrd/files/olsrd.init
@@ -1,7 +1,13 @@
 #!/bin/sh /etc/rc.common
 # Copyright (C) 2008  Alina Friedrichsen <x-alina@gmx.net>
 # Special thanks to bittorf wireless ))
+# 20101212 - some more (ugly) hacks added for 6and4 by soma
+
 START=65
+BIN=/usr/sbin/olsrd
+CONF=/var/etc/olsrd.conf
+PID=/var/run/olsrd.pid
+PID6=/var/run/olsrd.ipv6.pid
 
 OLSRD_OLSRD_SCHEMA='ignore:internal config_file:internal DebugLevel=0 AllowNoInt=yes'
 OLSRD_IPCCONNECT_SCHEMA='ignore:internal Host:list Net:list2'
@@ -319,7 +325,6 @@ olsrd_write_olsrd() {
 	local cfg="$1"
 	validate_varname "$cfg" || return 0
 	local ignore
-	local ipversion
 
 	config_get_bool ignore "$cfg" ignore 0
 	[ "$ignore" -ne 0 ] && return 0
@@ -456,9 +461,15 @@ olsrd_write_loadplugin() {
 				fi
 			fi
 
-			config_get latlon_file "$cfg" latlon_file
+			for f in latlon_file hosts_file services_file resolv_file macs_file; do
+				config_get $f "$cfg" $f
+			done
+
 			[ -z "$latlon_file" ] && config_set "$cfg" latlon_file '/var/run/latlon.js'
 		;;
+		olsrd_watchdog.*)
+			config_get wd_file "$cfg" file
+		;;
 	esac
 
 	echo -n "${N}LoadPlugin \"$library\"${N}{"
@@ -593,20 +604,51 @@ start() {
 	local bindv6only='0'
 	if [ "$OLSRD_IPVERSION_6AND4" -ne 0 ]; then
 		bindv6only="$(sysctl -n net.ipv6.bindv6only)"
-		sysctl -w net.ipv6.bindv6only=1
-		sed -e 's/^\t\t[0-9.]*[ ][0-9.]*$//' < "$OLSRD_CONFIG_FILE" > /var/etc/olsrd.conf.ipv6
-		sed -i '/[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}/d' /var/etc/olsrd.conf.ipv6
-		olsrd -f /var/etc/olsrd.conf.ipv6 -nofork < /dev/null > /dev/null &
+		sysctl -w net.ipv6.bindv6only=1 > /dev/null
+		sed -e '/[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}/d' < "$OLSRD_CONFIG_FILE" > /var/etc/olsrd.conf.ipv6
+		sed -e 's/^IpVersion[ ][ ]*6$/IpVersion 4/' -e 's/^\t\t[A-Fa-f0-9.:]*[:][A-Fa-f0-9.:]*[ ][0-9]*$//' -e 's/ether/mesh/' < "$OLSRD_CONFIG_FILE" > /var/etc/olsrd.conf.ipv4
+		rm "$OLSRD_CONFIG_FILE"
+
+		# some filenames should get the suffix .ipv6
+		for file in $latlon_file $hosts_file $services_file $resolv_file $macs_file $wd_file;do
+			f=$(echo $file|sed 's/\//\\\//g')
+			sed -i "s/$f/$f.ipv6/g" /var/etc/olsrd.conf.ipv6
+		done
+
+		if ([ -s $PID6 ] && kill -0 $(cat $PID6) 2>&-); then
+			echo "Error: There already is a IPv6 instance of olsrd running ($(cat $PID6)), not starting."
+		else
+			start-stop-daemon -q -b -m -p $PID6 -x $BIN -S -- -f /var/etc/olsrd.conf.ipv6 -nofork
+		fi
+
+                if ([ -s $PID ] && kill -0 $(cat $PID) 2>&-); then
+				echo "Error: There already is a IPv4 instance of olsrd running ($(cat $PID)), not starting."
+		else
+			start-stop-daemon -q -b -m -p $PID -x $BIN -S -- -f /var/etc/olsrd.conf.ipv4 -nofork
+		fi
 
-		sed -e 's/^IpVersion[ ][ ]*6$/IpVersion 4/' -e 's/^\t\t[A-Fa-f0-9.:]*[:][A-Fa-f0-9.:]*[ ][0-9]*$//' < "$OLSRD_CONFIG_FILE" > /var/etc/olsrd.conf.ipv4
-		olsrd -f /var/etc/olsrd.conf.ipv4 -nofork < /dev/null > /dev/null &
 		sleep 3
-		sysctl -w net.ipv6.bindv6only="$bindv6only"
+		sysctl -w net.ipv6.bindv6only="$bindv6only" > /dev/null
+
 	else
-		olsrd -f "$OLSRD_CONFIG_FILE" -nofork < /dev/null > /dev/null &
+
+		if [ "$ipversion" = "6" ]; then
+			sed -i '/[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}/d' "$OLSRD_CONFIG_FILE"
+		fi
+
+		### check for running instance
+		if ([ -s $PID ] && kill -0 $(cat $PID) 2>&-); then
+			echo "Error: There already is a instance of olsrd running ($(cat $PID)), exiting now."
+			return 1
+		else
+			start-stop-daemon -q -b -m -p $PID -x $BIN -S -- -f "$OLSRD_CONFIG_FILE" -nofork
+		fi
 	fi
 }
 
 stop() {
-	killall olsrd
+	start-stop-daemon -q -p $PID -x $BIN -K
+	start-stop-daemon -q -p $PID6 -x $BIN -K
+	test -s $PID && rm $PID
+	test -s $PID6 && rm $PID6
 }
