diff --exclude .svn -Nru package/crda/Makefile package/crda/Makefile
--- package/crda/Makefile	2012-08-17 15:40:44.901971001 +0200
+++ package/crda/Makefile	2012-08-18 14:47:32.615458002 +0200
@@ -1,5 +1,5 @@
 #
-# Copyright (C) 2009 OpenWrt.org
+# Copyright (C) 2009-2012 OpenWrt.org
 #
 # This is free software, licensed under the GNU General Public License v2.
 # See /LICENSE for more information.
@@ -10,11 +10,10 @@
 
 PKG_NAME:=crda
 PKG_RELEASE:=1
-PKG_VERSION:=1.1.1
+PKG_VERSION:=1.1.2
 PKG_SOURCE_URL:=http://wireless.kernel.org/download/crda
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
-PKG_MD5SUM:=5fc77af68b3e21736b8ef2f8b061c810
-PKG_BUILD_DEPENDS:=mac80211
+PKG_MD5SUM:=5226f65aebacf94baaf820f8b4e06df4
 
 PKG_REGULATORY_NAME:=regulatory
 PKG_REGULATORY_VERSION:=2011.04.28
@@ -28,7 +27,7 @@
   SECTION:=net
   CATEGORY:=Network
   TITLE:=Central Regulatory Domain Agent (CRDA)
-  DEPENDS:=+libnl-tiny @!LINUX_2_4
+  DEPENDS:=+libnl-tiny
   URL:=http://wireless.kernel.org/en/developers/Regulatory/CRDA
 endef
 
diff --exclude .svn -Nru package/crda/patches/101-make_crypto_use_optional.patch package/crda/patches/101-make_crypto_use_optional.patch
--- package/crda/patches/101-make_crypto_use_optional.patch	2012-08-17 15:40:44.901971001 +0200
+++ package/crda/patches/101-make_crypto_use_optional.patch	2012-08-18 14:47:32.615458002 +0200
@@ -1,6 +1,6 @@
 --- a/Makefile
 +++ b/Makefile
-@@ -32,7 +32,9 @@ LDLIBS += `pkg-config --libs openssl`
+@@ -35,7 +35,9 @@ LDLIBS += `pkg-config --libs openssl`
  
  reglib.o: keys-ssl.c
  
