diff --git a/net/olsrd/Makefile b/net/olsrd/Makefile
index e176185..e25d408 100644
--- a/net/olsrd/Makefile
+++ b/net/olsrd/Makefile
@@ -8,13 +8,13 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=olsrd
-PKG_VERSION:=0.6.3
-PKG_RELEASE:=3
+PKG_VERSION:=0.6.5.2
+PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
 PKG_SOURCE_URL:=http://www.olsr.org/releases/0.6
 
-PKG_MD5SUM:=b28ed2e3dc6c529ab690f1c805ad32a2
+PKG_MD5SUM:=190e3c763656f71d3ca6868db7658f43
 PKG_BUILD_PARALLEL:=1
 
 include $(INCLUDE_DIR)/package.mk
@@ -119,12 +119,6 @@ define Package/olsrd-mod-secure
   TITLE:=Message signing plugin to secure routing domain
 endef
 
-define Package/olsrd-mod-tas
-  $(call Package/olsrd/template)
-  DEPENDS:=olsrd
-  TITLE:=tas - tiny appliation server programming example
-endef
-
 define Package/olsrd-mod-txtinfo
   $(call Package/olsrd/template)
   DEPENDS:=olsrd
@@ -147,7 +141,7 @@ MAKE_FLAGS+= \
 	DESTDIR="$(PKG_INSTALL_DIR)" \
 	STRIP="true" \
 	INSTALL_LIB="true" \
-	SUBDIRS="arprefresh bmf dot_draw dyn_gw dyn_gw_plain httpinfo jsoninfo mdns nameservice p2pd pgraph quagga secure tas txtinfo watchdog"
+	SUBDIRS="arprefresh bmf dot_draw dyn_gw dyn_gw_plain httpinfo jsoninfo mdns nameservice p2pd pgraph quagga secure txtinfo watchdog"
 
 define Build/Compile
 	$(call Build/Compile/Default,all)
@@ -230,11 +224,6 @@ define Package/olsrd-mod-secure/install
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/lib/secure/olsrd_secure.so.* $(1)/usr/lib/
 endef
 
-define Package/olsrd-mod-tas/install
-	$(INSTALL_DIR) $(1)/usr/lib
-	$(INSTALL_BIN) $(PKG_BUILD_DIR)/lib/tas/olsrd_tas.so.* $(1)/usr/lib/
-endef
-
 define Package/olsrd-mod-txtinfo/install
 	$(INSTALL_DIR) $(1)/usr/lib
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/lib/txtinfo/olsrd_txtinfo.so.* $(1)/usr/lib/
@@ -259,6 +248,5 @@ $(eval $(call BuildPackage,olsrd-mod-p2pd))
 $(eval $(call BuildPackage,olsrd-mod-pgraph))
 $(eval $(call BuildPackage,olsrd-mod-quagga))
 $(eval $(call BuildPackage,olsrd-mod-secure))
-$(eval $(call BuildPackage,olsrd-mod-tas))
 $(eval $(call BuildPackage,olsrd-mod-txtinfo))
 $(eval $(call BuildPackage,olsrd-mod-watchdog))
diff --git a/net/olsrd/files/olsrd.config b/net/olsrd/files/olsrd.config
index 9a19eb9..9f173ee 100644
--- a/net/olsrd/files/olsrd.config
+++ b/net/olsrd/files/olsrd.config
@@ -22,5 +22,7 @@ config LoadPlugin
 	option library 'olsrd_txtinfo.so.0.1'
 	option accept '0.0.0.0'
 
-config Interface
-	list interface 'wlan'
+config LoadPlugin
+	option library 'olsrd_jsoninfo.so.0.0'
+	option accept '0.0.0.0'
+
diff --git a/net/olsrd/files/olsrd.init b/net/olsrd/files/olsrd.init
index 7578316..434aefe 100644
--- a/net/olsrd/files/olsrd.init
+++ b/net/olsrd/files/olsrd.init
@@ -662,8 +662,8 @@ start() {
 	if [ "$OLSRD_IPVERSION_6AND4" -ne 0 ]; then
 		bindv6only="$(sysctl -n net.ipv6.bindv6only)"
 		sysctl -w net.ipv6.bindv6only=1 > /dev/null
-		sed -e '/[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}/d' < "$OLSRD_CONFIG_FILE" > /var/etc/olsrd.conf.ipv6
-		sed -e 's/^IpVersion[ ][ ]*6$/IpVersion 4/' -e 's/^\t\t[A-Fa-f0-9.:]*[:][A-Fa-f0-9.:]*[ ][0-9]*$//' < "$OLSRD_CONFIG_FILE" > /var/etc/olsrd.conf.ipv4
+		sed -e 's/0.0.0.0/::/' -e '/[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}/d' -e 's/etx_ff/etx_ffeth/' -e '/NatThreshold/d' -e '/RtTable/d' < "$OLSRD_CONFIG_FILE" > /var/etc/olsrd.conf.ipv6
+		sed -e 's/^IpVersion[ ][ ]*6$/IpVersion 4/' -e 's/^\tPlParam \"[A-Fa-f0-9.:]*[:][A-Fa-f0-9.:].*//' -e 's/^\t\t[A-Fa-f0-9.:]*[:][A-Fa-f0-9.:]*[ ][0-9]*$//' -e 's/ether/mesh/' < "$OLSRD_CONFIG_FILE" > /var/etc/olsrd.conf.ipv4
 		rm $OLSRD_CONFIG_FILE
 
 		# some filenames should get the suffix .ipv6
diff --git a/net/olsrd/patches/100-olsrd-fix-stack-corruption-in-net_output.patch b/net/olsrd/patches/100-olsrd-fix-stack-corruption-in-net_output.patch
deleted file mode 100644
index 77c2323..0000000
--- a/net/olsrd/patches/100-olsrd-fix-stack-corruption-in-net_output.patch
+++ /dev/null
@@ -1,52 +0,0 @@
-From f4d250ad4fad5fcfe5b5feaac3f3e121adef3fba Mon Sep 17 00:00:00 2001
-From: Jo-Philipp Wich <jow@openwrt.org>
-Date: Fri, 22 Jun 2012 03:17:59 +0200
-Subject: [PATCH] olsrd: fix stack corruption in net_output()
-
-The net_output() function indirectly uses the stack variables dst and dst6
-outside of the scope they're declared in, this might leads to olsr_sendto()
-being called with a corrupted destination sockaddr_in.
-
-This failure condition can be observed in the log, olsrd will continuosly
-print "sendto(v4): Invalid Argument" or a similar message. On ARM it has been
-reported to result in "Unsupported Address Family".
-
-This bug became apparant on a custon OpenWrt x86_64 uClibc target using the
-Linaro GCC 4.7-2012.04 compiler, it has been reported for an unspecified ARM
-target as well.
-
-The offending code seems to be unchanged since 2008 and it does not cause
-issues on 32bit systems and/or with older (Linaro) GCC versions, but the
-compiler used in our tests seems to perform more aggressive optimizations
-leading to a stack corruption.
----
- src/net_olsr.c |    4 ++--
- 1 files changed, 2 insertions(+), 2 deletions(-)
-
---- a/src/net_olsr.c
-+++ b/src/net_olsr.c
-@@ -336,6 +336,8 @@ net_output(struct interface *ifp)
- {
-   struct sockaddr_in *sin = NULL;
-   struct sockaddr_in6 *sin6 = NULL;
-+  struct sockaddr_in dst;
-+  struct sockaddr_in6 dst6;
-   struct ptf *tmp_ptf_list;
-   union olsr_packet *outmsg;
-   int retval;
-@@ -354,7 +356,6 @@ net_output(struct interface *ifp)
-   outmsg->v4.olsr_packlen = htons(ifp->netbuf.pending);
- 
-   if (olsr_cnf->ip_version == AF_INET) {
--    struct sockaddr_in dst;
-     /* IP version 4 */
-     sin = (struct sockaddr_in *)&ifp->int_broadaddr;
- 
-@@ -365,7 +366,6 @@ net_output(struct interface *ifp)
-     if (sin->sin_port == 0)
-       sin->sin_port = htons(olsr_cnf->olsrport);
-   } else {
--    struct sockaddr_in6 dst6;
-     /* IP version 6 */
-     sin6 = (struct sockaddr_in6 *)&ifp->int6_multaddr;
-     /* Copy sin */
