diff --git a/net/net-snmp/Makefile b/net/net-snmp/Makefile
index 8fd1c4f..24b256a 100644
--- a/net/net-snmp/Makefile
+++ b/net/net-snmp/Makefile
@@ -253,7 +253,7 @@ endef
 
 define Package/snmp-utils/install
 	$(INSTALL_DIR) $(1)/usr/bin
-	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/snmp{get,set,status,test,trap,walk} $(1)/usr/bin/
+	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/snmp{get,getnext,set,status,test,trap,walk} $(1)/usr/bin/
 endef
 
 $(eval $(call BuildPackage,libnetsnmp))
