Index: libs/libevent2/Makefile
===================================================================
--- libs/libevent2/Makefile	(revision 25247)
+++ libs/libevent2/Makefile	(revision 27157)
@@ -9,5 +9,5 @@
 
 PKG_NAME:=libevent2
-PKG_VERSION:=2.0.10
+PKG_VERSION:=2.0.12
 PKG_RELEASE:=1
 
@@ -15,5 +15,5 @@
 PKG_SOURCE:=libevent-$(PKG_VERSION)-stable.tar.gz
 PKG_SOURCE_URL:=http://www.monkey.org/~provos/
-PKG_MD5SUM:=a37401d26cbbf28185211d582741a3d4
+PKG_MD5SUM:=42986228baf95e325778ed328a93e070
 
 PKG_FIXUP:=libtool
@@ -23,12 +23,12 @@
 include $(INCLUDE_DIR)/package.mk
 
-define Package/libevent2
+define Package/libevent2/Default
   SECTION:=libs
   CATEGORY:=Libraries
-  TITLE:=Event notification library (version 2.0)
+  TITLE:=Event notification
   URL:=http://www.monkey.org/~provos/libevent/
 endef
 
-define Package/libevent2/description
+define Package/libevent2/Default/description
 	The libevent API provides a mechanism to execute a callback function
 	when a specific event occurs on a file descriptor or after a timeout
@@ -40,4 +40,66 @@
 	and then add or remove events dynamically without having to change
 	the event loop.
+endef
+
+define Package/libevent2
+  $(call Package/libevent2/Default)
+  TITLE+=  library (version 2.0)
+endef
+
+define Package/libevent2/description
+	$(call Package/libevent2/Default/description)
+
+	This package contains the libevent shared library historically
+	containing both the core & extra libraries.
+endef
+
+define Package/libevent2-core
+  $(call Package/libevent2/Default)
+  TITLE+= core library (version 2.0)
+endef
+
+define Package/libevent2-core/description
+	$(call Package/libevent2/Default/description)
+
+	This package contains the libevent core shared library for the event,
+	buffer & utility functions.
+endef
+
+define Package/libevent2-extra
+  $(call Package/libevent2/Default)
+  TITLE+= extra library (version 2.0)
+endef
+
+define Package/libevent2-extra/description
+	$(call Package/libevent2/Default/description)
+
+	This package contains the libevent extra shared library for specific
+	protocols including HTTP, DNS & RPC.
+endef
+
+define Package/libevent2-openssl
+  $(call Package/libevent2/Default)
+  TITLE+= OpenSSL library (version 2.0)
+  DEPENDS:=+libopenssl
+endef
+
+define Package/libevent2-openssl/description
+	$(call Package/libevent2/Default/description)
+
+	This package contains the libevent OpenSSL shared library for encrypted
+	bufferevents.
+endef
+
+define Package/libevent2-pthreads
+  $(call Package/libevent2/Default)
+  TITLE+= Pthreads library (version 2.0)
+  DEPENDS:=+libpthread
+endef
+
+define Package/libevent2-pthreads/description
+	$(call Package/libevent2/Default/description)
+
+	This package contains the libevent Pthreads shared library for
+	threading & locking.
 endef
 
@@ -55,8 +117,8 @@
 	$(CP) $(PKG_INSTALL_DIR)/usr/include/* $(1)/usr/include/
 	$(INSTALL_DIR) $(1)/usr/lib
-	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libevent.{la,a,so} $(1)/usr/lib/
-	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libevent-2.0.so* $(1)/usr/lib/
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libevent*.{la,a,so} $(1)/usr/lib/
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libevent*-2.0.so* $(1)/usr/lib/
 	$(INSTALL_DIR) $(1)/usr/lib/pkgconfig
-	$(CP) $(PKG_INSTALL_DIR)/usr/lib/pkgconfig/libevent.pc $(1)/usr/lib/pkgconfig
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/pkgconfig/libevent*.pc $(1)/usr/lib/pkgconfig/
 endef
 
@@ -66,3 +128,27 @@
 endef
 
+define Package/libevent2-core/install
+	$(INSTALL_DIR) $(1)/usr/lib
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libevent_core-2.0.so.* $(1)/usr/lib/
+endef
+
+define Package/libevent2-extra/install
+	$(INSTALL_DIR) $(1)/usr/lib
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libevent_extra-2.0.so.* $(1)/usr/lib/
+endef
+
+define Package/libevent2-openssl/install
+	$(INSTALL_DIR) $(1)/usr/lib
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libevent_openssl-2.0.so.* $(1)/usr/lib/
+endef
+
+define Package/libevent2-pthreads/install
+	$(INSTALL_DIR) $(1)/usr/lib
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libevent_pthreads-2.0.so.* $(1)/usr/lib/
+endef
+
 $(eval $(call BuildPackage,libevent2))
+$(eval $(call BuildPackage,libevent2-core))
+$(eval $(call BuildPackage,libevent2-extra))
+$(eval $(call BuildPackage,libevent2-openssl))
+$(eval $(call BuildPackage,libevent2-pthreads))
