Index: package/uhttpd/src/uhttpd-mimetypes.h
===================================================================
--- package/uhttpd/src/uhttpd-mimetypes.h	(revision 32877)
+++ package/uhttpd/src/uhttpd-mimetypes.h	(working copy)
@@ -78,6 +78,8 @@
 
 	{ "pac",		"application/x-ns-proxy-autoconfig" },
 	{ "wpad.dat",	"application/x-ns-proxy-autoconfig" },
+	{ "appcache", "text/cache-manifest" },
+	{ "manifest", "text/cache-manifest" },
 
 	{ NULL, NULL }
 };
