Index: ipv6/radvd/files/radvd.init
===================================================================
--- ipv6/radvd/files/radvd.init	(Revision 25315)
+++ ipv6/radvd/files/radvd.init	(Arbeitskopie)
@@ -78,7 +78,7 @@
 
 	ip6addr=$(LANG=C ip -f inet6 -- addr show dev "$ifname" 2> /dev/null | sed -n -e 's/^  *inet6 \([A-Fa-f0-9.:]*[/][0-9]*\) scope '"$iproute2_scope"' $/\1/p' | head -n 1)
 	if [ -z "$ip6addr" ]; then
-		ip6addr=$(LANG=C ifconfig "$ifname" 2> /dev/null | sed -n -e 's/^  *inet6 addr: \([A-Fa-f0-9.:]*[/][0-9]*\) Scope:'"$ifconfig_scope"'$/\1/p' | tail -n 1)
+		ip6addr=$(LANG=C ifconfig "$ifname" 2> /dev/null | sed -n -e 's/^  *inet6 addr: \([A-Fa-f0-9.:]*[/][0-9]*\) Scope:'"$ifconfig_scope"'$/\1/p' | head -n 1)
 		[ -z "$ip6addr" ] && return 1
 	fi
 
@@ -406,7 +406,7 @@
 
 	sysctl -w net.ipv6.conf.all.forwarding=1 > /dev/null 2> /dev/null
 
-	radvd -C "$RADVD_CONFIG_FILE" -m stderr_syslog -p /var/run/radvd.pid
+	radvd -C "$RADVD_CONFIG_FILE" -m none -p /var/run/radvd.pid
 }
 
 stop() {
