diff --git a/nat46/Makefile b/nat46/Makefile
index d1dfcd9..1007337 100644
--- a/nat46/Makefile
+++ b/nat46/Makefile
@@ -12,7 +12,7 @@ PKG_NAME:=nat46
 PKG_VERSION:=1
 PKG_RELEASE:=$(PKG_SOURCE_VERSION)
 
-PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
 PKG_SOURCE_URL:=https://github.com/ayourtch/nat46.git
 PKG_SOURCE_PROTO:=git
