Index: target/linux/ar71xx/base-files/etc/defconfig/routerstation-pro/network
===================================================================
--- target/linux/ar71xx/base-files/etc/defconfig/routerstation-pro/network	(Revision 26720)
+++ target/linux/ar71xx/base-files/etc/defconfig/routerstation-pro/network	(Arbeitskopie)
@@ -5,16 +5,12 @@
 	option netmask	255.0.0.0
 
 config interface lan
-	option ifname	eth1
+	option ifname	'eth1 eth0'
 	option type	bridge
 	option proto	static
 	option ipaddr	192.168.1.1
 	option netmask	255.255.255.0
 
-config interface wan
-	option ifname	eth0
-	option proto	dhcp
-
 config switch
         option name     eth1
         option reset    1
