diff --git a/target/linux/at91/image/Makefile b/target/linux/at91/image/Makefile
index 97723bb..45f9376 100644
--- a/target/linux/at91/image/Makefile
+++ b/target/linux/at91/image/Makefile
@@ -18,12 +18,13 @@ define Build/Compile
 endef
 
 define Image/Prepare
+	cp $(LINUX_DIR)/arch/arm/boot/Image $(KDIR)/Image
 endef
 
 define Image/BuildKernel
 	mkdir -p $(BIN_DIR)
 	mkimage -A arm -T kernel -C none -a 0x20008000 -e 0x20008000 -n linux-2.6 \
-		-d $(LINUX_DIR)/arch/arm/boot/Image $(BIN_DIR)/$(IMG_PREFIX)-uImage
+		-d $(KDIR)/Image $(BIN_DIR)/$(IMG_PREFIX)-uImage
 	if [ $(CONFIG_FLEXIBITY_ROOT) ]; then \
 		$(INSTALL_BIN) $(BIN_DIR)/$(IMG_PREFIX)-uImage $(TARGET_DIR)/uImage ; \
 	fi
