diff --git a/target/linux/brcm63xx/profiles/200-GW6X00.mk b/target/linux/brcm63xx/profiles/200-GW6X00.mk
index b1c516a..418c88e 100644
--- a/target/linux/brcm63xx/profiles/200-GW6X00.mk
+++ b/target/linux/brcm63xx/profiles/200-GW6X00.mk
@@ -7,7 +7,8 @@
 
 define Profile/GW6X00
   NAME:=Tecom GW6X00
-  PACKAGES:=kmod-brcm-wl kmod-usb-core kmod-usb-ohci kmod-usb-storage \
+  PACKAGES:=kmod-brcm-wl kmod-usb-core kmod-usb-ohci kmod-usb-serial \
+	kmod-usb-serial-option comgt kmod-usb-storage \
 	kmod-fs-ext4 kmod-nls-cp437 kmod-nls-iso8859-1 e2fsprogs \
 	kmod-bcm63xx-spi \
 	kmod-ipt-nathelper-extra wlc
