diff --git a/target/linux/x86/kvm_guest/target.mk b/target/linux/x86/kvm_guest/target.mk
index 7e48f64..80c3e6f 100644
--- a/target/linux/x86/kvm_guest/target.mk
+++ b/target/linux/x86/kvm_guest/target.mk
@@ -1,2 +1,4 @@
 BOARDNAME:=KVM Guest
-DEFAULT_PACKAGES += kmod-virtio-balloon kmod-virtio-net kmod-virtio-random
+DEFAULT_PACKAGES += kmod-virtio-balloon kmod-virtio-net kmod-virtio-random \
+	kmod-usb2 kmod-pcnet32 kmod-vmxnet3 kmod-e1000
+
