diff --git a/package/brcm2708-gpu-fw/Makefile b/package/brcm2708-gpu-fw/Makefile
index 0b79cd1..11b7b54 100644
--- a/package/brcm2708-gpu-fw/Makefile
+++ b/package/brcm2708-gpu-fw/Makefile
@@ -40,5 +40,9 @@ define Build/InstallDev
 	$(CP) $(PKG_BUILD_DIR)/* $(BUILD_DIR)/brcm2708-gpu-fw-boot
 endef
 
+define Package/brcm2708-gpu-fw/install
+  # nothing to do
+endef
+
 $(eval $(call BuildPackage,brcm2708-gpu-fw))
 
