diff --git a/target/linux/brcm47xx/profiles/PS-1208MFG.mk b/target/linux/brcm47xx/profiles/PS-1208MFG.mk
index 5d7388b..f6a30a4 100644
--- a/target/linux/brcm47xx/profiles/PS-1208MFG.mk
+++ b/target/linux/brcm47xx/profiles/PS-1208MFG.mk
@@ -7,7 +7,7 @@
 
 define Profile/Ps1208mfg
   NAME:=Edimax PS-1208MFG
-  PACKAGES:=-firewall -dropbear -dnsmasq -mtd -ppp -wpad-mini block-mount kmod-usb-storage kmod-usb2 kmod-usb-ohci -iptables -kmod-switch kmod-fs-ext4 block-extroot
+  PACKAGES:=-firewall -dropbear -dnsmasq -mtd -ppp -wpad-mini block-mount kmod-usb-storage kmod-usb2 kmod-usb-ohci -iptables -kmod-switch kmod-fs-ext4 block-mount
 endef
 
 define Profile/Ps1208mfg/Description
