diff --git a/target/linux/ixp4xx/generic/profiles/200-NSLU2.mk b/target/linux/ixp4xx/generic/profiles/200-NSLU2.mk
index 523bda4..ad417e0 100644
--- a/target/linux/ixp4xx/generic/profiles/200-NSLU2.mk
+++ b/target/linux/ixp4xx/generic/profiles/200-NSLU2.mk
@@ -8,8 +8,9 @@
 define Profile/NSLU2
   NAME:=Linksys NSLU2
   PACKAGES:=-wpad-mini -kmod-ath5k kmod-scsi-core \
-	kmod-usb-core kmod-usb-ohci kmod-usb2 kmod-usb-storage \
-	kmod-fs-ext2 kmod-fs-ext3
+	kmod-usb-core kmod-usb-ohci kmod-usb2 kmod-usb-serial \
+	kmod-usb-serial-option comgt kmod-usb-storage \
+	kmod-fs-ext4 kmod-nls-cp437 kmod-nls-iso8859-1 e2fsprogs
 endef
 
 define Profile/NSLU2/Description
diff --git a/target/linux/ixp4xx/generic/profiles/300-NAS100d.mk b/target/linux/ixp4xx/generic/profiles/300-NAS100d.mk
index 36a544b..5220d43 100644
--- a/target/linux/ixp4xx/generic/profiles/300-NAS100d.mk
+++ b/target/linux/ixp4xx/generic/profiles/300-NAS100d.mk
@@ -10,8 +10,9 @@ define Profile/NAS100d
   PACKAGES:=kmod-ath5k \
 	kmod-scsi-core \
 	kmod-ata-core kmod-ata-artop \
-	kmod-usb-core kmod-usb2 kmod-usb-storage \
-	kmod-fs-ext2 kmod-fs-ext3
+	kmod-usb-core kmod-usb2 kmod-usb-serial \
+	kmod-usb-serial-option comgt kmod-usb-storage \
+	kmod-fs-ext4 kmod-nls-cp437 kmod-nls-iso8859-1 e2fsprogs
 endef
 
 define Profile/NAS100d/Description
diff --git a/target/linux/ixp4xx/generic/profiles/400-DSMG600RevA.mk b/target/linux/ixp4xx/generic/profiles/400-DSMG600RevA.mk
index 6e6afba..0f1e5bc 100644
--- a/target/linux/ixp4xx/generic/profiles/400-DSMG600RevA.mk
+++ b/target/linux/ixp4xx/generic/profiles/400-DSMG600RevA.mk
@@ -11,8 +11,9 @@ define Profile/DSMG600RevA
 	kmod-ath5k \
 	kmod-scsi-core \
 	kmod-ata-core kmod-ata-artop \
-	kmod-usb-core kmod-usb-uhci kmod-usb2 kmod-usb-storage \
-	kmod-fs-ext2 kmod-fs-ext3
+	kmod-usb-core kmod-usb-uhci kmod-usb2 kmod-usb-serial \
+	kmod-usb-serial-option comgt kmod-usb-storage \
+	kmod-fs-ext4 kmod-nls-cp437 kmod-nls-iso8859-1 e2fsprogs
 endef
 
 define Profile/DSMG600RevA/Description
diff --git a/target/linux/ixp4xx/generic/profiles/500-USR8200.mk b/target/linux/ixp4xx/generic/profiles/500-USR8200.mk
index 99cda13..0e5d3c7 100644
--- a/target/linux/ixp4xx/generic/profiles/500-USR8200.mk
+++ b/target/linux/ixp4xx/generic/profiles/500-USR8200.mk
@@ -8,8 +8,9 @@
 define Profile/USR8200
   NAME:=USRobotics USR8200
   PACKAGES:=-wpad-mini kmod-scsi-core \
-	kmod-usb-core kmod-usb-uhci kmod-usb2 kmod-usb-storage \
-	kmod-fs-ext2 kmod-fs-ext3 \
+	kmod-usb-core kmod-usb-uhci kmod-usb2 kmod-usb-serial \
+	kmod-usb-serial-option comgt kmod-usb-storage \
+	kmod-fs-ext4 kmod-nls-cp437 kmod-nls-iso8859-1 e2fsprogs\
 	kmod-ieee1394 kmod-ohci1394 kmod-sbp2
 endef
 
