Index: a/package/brcm2708-gpu-fw/Makefile
===================================================================
--- a/package/brcm2708-gpu-fw/Makefile	(revision 35471)
+++ b/package/brcm2708-gpu-fw/Makefile	(working copy)
@@ -8,8 +8,8 @@
 include $(INCLUDE_DIR)/kernel.mk
 
 PKG_NAME:=brcm2708-gpu-fw
-PKG_REV:=29ce6bcf278e7e79cd8a2b84c6aa9db9864ca91d
-PKG_VERSION:=20120529
+PKG_REV:=4bec05a5724bf0a224aab475a05225bdfd7633bf
+PKG_VERSION:=20130428
 PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
