diff --git a/target/linux/ar71xx/generic/profiles/ubnt.mk b/target/linux/ar71xx/generic/profiles/ubnt.mk
index dc26956..581493f 100644
--- a/target/linux/ar71xx/generic/profiles/ubnt.mk
+++ b/target/linux/ar71xx/generic/profiles/ubnt.mk
@@ -7,7 +7,7 @@
 
 define Profile/UBNTRS
 	NAME:=Ubiquiti RouterStation
-	PACKAGES:=kmod-usb-core kmod-usb-ohci kmod-usb2
+	PACKAGES:=kmod-usb-core kmod-usb-ohci kmod-usb2 kmod-ath5k
 endef
 
 define Profile/UBNTRS/Description
@@ -18,7 +18,7 @@ $(eval $(call Profile,UBNTRS))
 
 define Profile/UBNTRSPRO
 	NAME:=Ubiquiti RouterStation Pro
-	PACKAGES:=kmod-usb-core kmod-usb-ohci kmod-usb2
+	PACKAGES:=kmod-usb-core kmod-usb-ohci kmod-usb2 kmod-ath5k
 endef
 
 define Profile/UBNTRSPRO/Description
