diff --git a/target/linux/ixp4xx/image/Makefile b/target/linux/ixp4xx/image/Makefile
index 7a69998..1777803 100644
--- a/target/linux/ixp4xx/image/Makefile
+++ b/target/linux/ixp4xx/image/Makefile
@@ -10,12 +10,12 @@ include $(INCLUDE_DIR)/image.mk
 ifdef CONFIG_PACKAGE_apex
   define Image/Build/Linksys
 	BIN_DIR=$(BIN_DIR) $(TOPDIR)/scripts/slugimage.pl \
-		-L $(BIN_DIR)/apex/apex-$(2)-armeb.bin \
+		-L $(KDIR)/apex-$(2)-armeb.bin \
 		-k $(BIN_DIR)/openwrt-$(2)-zImage \
 		-r rootfs:$(BIN_DIR)/$(IMG_PREFIX)-$(1).img \
 		-p -o $(BIN_DIR)/openwrt-$(2)-$(1).bin
 	BIN_DIR=$(BIN_DIR) $(TOPDIR)/scripts/slugimage.pl \
-		-F -L $(BIN_DIR)/apex/apex-$(2)-16mb-armeb.bin \
+		-F -L $(KDIR)/apex-$(2)-16mb-armeb.bin \
 		-k $(BIN_DIR)/openwrt-$(2)-zImage \
 		-r rootfs:$(BIN_DIR)/$(IMG_PREFIX)-$(1).img \
 		-p -o $(BIN_DIR)/openwrt-$(2)-$(1)-16mb.bin
@@ -31,8 +31,17 @@ define Image/Build/Freecom
 	rm -f $(TARGET_DIR)/zImage
 endef
 
+define Image/Build/Avila
+	-sh $(TOPDIR)/scripts/combined-image.sh \
+		"$(BIN_DIR)/openwrt-avila-zImage" \
+		"$(BIN_DIR)/$(IMG_PREFIX)-squashfs.img" \
+		"$(BIN_DIR)/$(IMG_PREFIX)-avila-squashfs-sysupgrade.bin"
+endef
+
 define Image/Prepare
 	cp $(LINUX_DIR)/arch/arm/boot/zImage $(KDIR)/zImage
+	mkdir -p $(KDIR)/apex
+	cp $(BIN_DIR)/apex/* $(KDIR)
 endef
 
 define Image/BuildKernel
@@ -59,6 +68,7 @@ define Image/Build/squashfs
 	dd if=$(KDIR)/root.$(1) of=$(BIN_DIR)/$(IMG_PREFIX)-$(1).img bs=131072 conv=sync
 	$(call Image/Build/Linksys,$(1),nslu2,$(1))
 	$(call Image/Build/Freecom,$(1),fsg3,$(1))
+	$(call Image/Build/Avila)
 endef
 
 $(eval $(call BuildImage))
