diff --git a/target/linux/au1000/au1500/profiles/MeshCube.mk b/target/linux/au1000/au1500/profiles/MeshCube.mk
index 3dff376..12ac878 100644
--- a/target/linux/au1000/au1500/profiles/MeshCube.mk
+++ b/target/linux/au1000/au1500/profiles/MeshCube.mk
@@ -8,7 +8,8 @@
 define Profile/MeshCube
   NAME:=MeshCube MTX-1
   PACKAGES:=kmod-ath5k kmod-hostap kmod-hostap-pci kmod-net-prism54 \
-	kmod-usb-core kmod-usb-ohci kmod-usb-storage kmod-fs-ext3 kmod-fs-vfat
+	kmod-usb-core kmod-usb-ohci kmod-usb-storage kmod-fs-ext4 kmod-fs-vfat \
+	kmod-usb-serial kmod-usb-serial-option comgt
 endef
 
 define Profile/MeshCube/Description
