Index: a/target/linux/brcm2708/base-files/etc/inittab
===================================================================
--- a/target/linux/brcm2708/base-files/etc/inittab	(revision 35471)
+++ b/target/linux/brcm2708/base-files/etc/inittab	(working copy)
@@ -1,5 +1,5 @@
 ::sysinit:/etc/init.d/rcS S boot
 ::shutdown:/etc/init.d/rcS K shutdown
 tts/0::askfirst:/bin/ash --login
-ttyAMA0::askfirst:/bin/ash --login
+#ttyAMA0::askfirst:/bin/ash --login
 tty1::askfirst:/bin/ash --login
