Index: package/base-files/files/etc/passwd
===================================================================
--- package/base-files/files/etc/passwd	(Revision 29641)
+++ package/base-files/files/etc/passwd	(Arbeitskopie)
@@ -1,4 +1,4 @@
-root:x:0:0:root:/root:/bin/ash
+root:$1$xhfrr5XE$O5SKxfQ0Z8ftW/nB.DDdu/:0:0:root:/root:/bin/ash
 daemon:*:1:1:daemon:/var:/bin/false
 ftp:*:55:55:ftp:/home/ftp:/bin/false
 network:*:101:101:network:/var:/bin/false
