Index: package/base-files/files/etc/hotplug.d/net/10-net
===================================================================
--- package/base-files/files/etc/hotplug.d/net/10-net	(Revision 21942)
+++ package/base-files/files/etc/hotplug.d/net/10-net	(Arbeitskopie)
@@ -14,6 +14,7 @@
 		ppp*|3g-*) return 0;;
 		ath*) return 0;;
 		wlan*) return 0;;
+		wl*) return 0;;
 	esac
 
 	scan_interfaces
Index: package/broadcom-wl/files/lib/wifi/broadcom.sh
===================================================================
--- package/broadcom-wl/files/lib/wifi/broadcom.sh	(Revision 21942)
+++ package/broadcom-wl/files/lib/wifi/broadcom.sh	(Arbeitskopie)
@@ -360,7 +360,7 @@
 	option disabled 1
 
 config wifi-iface
-	option device   wl${i}
+	option device   wlan${i}
 	option network	lan
 	option mode     ap
 	option ssid     OpenWrt${i#0}
