From: template <nomail@local>
Date: Fri, 6 Sep 2019 08:29:50 +0000
Subject: wifi_nolegacy_rates_by_default

diff --git a/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js b/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js
index c25acf40c4412e0bd66a10f32b1965d5d9095a4b..5ac35323daaf762b9e5e3be7ba91ddc989f31dab 100644
--- a/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js
+++ b/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js
@@ -883,7 +883,7 @@ return view.extend({
 					o.wifiNetwork = radioNet;
 
 					o = ss.taboption('advanced', form.Flag, 'legacy_rates', _('Allow legacy 802.11b rates'));
-					o.default = o.enabled;
+					o.default = o.disabled;
 
 					o = ss.taboption('advanced', form.Value, 'distance', _('Distance Optimization'), _('Distance to farthest network member in meters.'));
 					o.datatype = 'or(range(0,114750),"auto")';
