From: template <nomail@local>
Date: Fri, 6 Sep 2019 08:29:50 +0000
Subject: wifi_nolegacy_rates_by_default

diff --git a/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js b/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js
index c03b559cb380266f1b6fe8a556dcd5eb8dd0ab25..56b6f87c609dd7a56885368a7ab094607b1c4a74 100644
--- a/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js
+++ b/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js
@@ -896,7 +896,7 @@ return view.extend({
 					o.value('3', _('Very High'));
 
 					o = ss.taboption('advanced', form.Flag, 'legacy_rates', _('Allow legacy 802.11b rates'));
-					o.default = o.enabled;
+					o.default = o.disabled;
 
 					o = ss.taboption('advanced', form.Value, 'distance', _('Distance Optimization'), _('Distance to farthest network member in meters.'));
 					o.datatype = 'or(range(0,114750),"auto")';
