From: template <nomail@local>
Date: Fri, 6 Sep 2019 08:29:50 +0000
Subject: wifi_no_meshfwding_by_default

---


diff -ur openwrt.orig/feeds/luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js openwrt/feeds/luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js
--- luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js	2019-11-14 01:08:59.557322662 +0100
+++ luci/modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js	2019-11-14 01:15:06.295988745 +0100
@@ -812,7 +812,7 @@
 
 				o = ss.taboption('advanced', form.Flag, 'mesh_fwding', _('Forward mesh peer traffic'));
 				o.rmempty = false;
-				o.default = '1';
+				o.default = '0';
 				o.depends('mode', 'mesh');
 
 				o = ss.taboption('advanced', form.Value, 'mesh_rssi_threshold', _('RSSI threshold for joining'), _('0 = not using RSSI threshold, 1 = do not change driver default'));
-- 
2.20.1


