diff --git a/net/l3roamd/Makefile b/net/l3roamd/Makefile
index f27a3a3..7333e34 100644
--- a/net/l3roamd/Makefile
+++ b/net/l3roamd/Makefile
@@ -1,10 +1,10 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=l3roamd
-PKG_SOURCE_DATE:=2019-09-12
+PKG_SOURCE_DATE:=2019-09-18
 PKG_SOURCE_PROTO:=git
 PKG_SOURCE_URL=https://github.com/freifunk-gluon/l3roamd.git
-PKG_SOURCE_VERSION=0f864daa509d0cd17207df9ef42af0d5e4e5e370
+PKG_SOURCE_VERSION=10809f221a12c608b0ffe42c5bd90e274c31cdc4
 
 CMAKE_OPTIONS += -DCMAKE_BUILD_TYPE:STRING=MINSIZEREL
 #CMAKE_OPTIONS += -DCMAKE_BUILD_TYPE:STRING=Debug
