From: template <nomail@local>
Date: Fri, 6 Sep 2019 08:29:50 +0000
Subject: imagebuilder-custom-postinst-script

diff --git a/target/imagebuilder/files/Makefile b/target/imagebuilder/files/Makefile
index aeae98aacdb0094594b874f4e4c021cc09decf72..ee1bb9fc6f3652c2b41dd97cc0af89ff2e6ba26e 100644
--- a/target/imagebuilder/files/Makefile
+++ b/target/imagebuilder/files/Makefile
@@ -121,6 +121,9 @@ _call_image: staging_dir/host/.prereq-build
 	$(MAKE) package_reload
 	$(MAKE) package_install
 	$(MAKE) -s prepare_rootfs
+ifneq ($(CUSTOM_POSTINST_SCRIPT),)
+	(cd "$(TARGET_DIR)" ; bash $(CUSTOM_POSTINST_SCRIPT))
+endif
 	$(MAKE) -s build_image
 	$(MAKE) -s json_overview_image_info
 	$(MAKE) -s checksum
