From: template <nomail@local>
Date: Fri, 6 Sep 2019 08:29:50 +0000
Subject: hostapd_nolegacy_by_default

diff --git a/package/network/services/hostapd/files/hostapd.sh b/package/network/services/hostapd/files/hostapd.sh
index 79e71616dcfd5af0d2396ed1db2202836c80c3f0..4327a715d803b923fcbb647389980de95adef0a5 100644
--- a/package/network/services/hostapd/files/hostapd.sh
+++ b/package/network/services/hostapd/files/hostapd.sh
@@ -112,7 +112,7 @@ hostapd_prepare_device_config() {
 
 	set_default country_ie 1
 	set_default doth 1
-	set_default legacy_rates 1
+	set_default legacy_rates 0
 
 	[ "$hwmode" = "b" ] && legacy_rates=1
 
