From: template <nomail@local>
Date: Fri, 6 Sep 2019 08:29:50 +0000
Subject: banner-info

diff --git a/package/base-files/files/etc/banner b/package/base-files/files/etc/banner
index f3af3c014f20f1b1157bde1cadae64cab0189c9f..c91c7e89d0a8cbe7aaefa3fb7e8e997f16b2f29a 100644
--- a/package/base-files/files/etc/banner
+++ b/package/base-files/files/etc/banner
@@ -1,8 +1,20 @@
-  _______                     ________        __
- |       |.-----.-----.-----.|  |  |  |.----.|  |_
- |   -   ||  _  |  -__|     ||  |  |  ||   _||   _|
- |_______||   __|_____|__|__||________||__|  |____|
-          |__| W I R E L E S S   F R E E D O M
- -----------------------------------------------------
- %D %V, %C
+  _____        _  __             _
+ |  ___|      (_)/ _|           | |
+ | |_ _ __ ___ _| |_ _   _ _ __ | | __
+ |  _| '__/ _ \ |  _| | | | '_ \| |/ /
+ | | | | |  __/ | | | |_| | | | |   <
+ \_| |_|  \___|_|_|  \__,_|_| |_|_|\_\
+
+ Firmware %D (%V rev %C)
+ %P - %S
+ https://wiki.freifunk.net/Berlin:Firmware
+ https://github.com/freifunk-berlin
  -----------------------------------------------------
+
+ If you find bugs please report them at:
+
+   https://github.com/freifunk-berlin/firmware/issues
+
+ For questions write a mail to <berlin@berlin.freifunk.net>
+ or check https://berlin.freifunk.net/contact for our weekly meetings.
+
