diff --git a/target/linux/ath79/base-files/etc/board.d/01_leds b/target/linux/ath79/base-files/etc/board.d/01_leds
index 8b378f5df1..35cb74db28 100755
--- a/target/linux/ath79/base-files/etc/board.d/01_leds
+++ b/target/linux/ath79/base-files/etc/board.d/01_leds
@@ -158,6 +158,17 @@ tplink,archer-d50-v1)
 	ucidef_set_led_switch "wan_data" "WAN Data" "tp-link:white:internet" "switch0" "0x02" "" "tx rx"
 	ucidef_set_led_switch "wan_link" "WAN Link" "tp-link:white:wan" "switch0" "0x02" "" "link"
 	;;
+tplink,cpe210-v1|\
+tplink,cpe220-v2|\
+tplink,cpe510-v1)
+	ucidef_set_led_netdev "lan0" "LAN0" "tp-link:green:lan0" "eth1"
+	ucidef_set_led_switch "lan1" "LAN1" "tp-link:green:lan1" "switch0" "0x10"
+	ucidef_set_rssimon "wlan0" "200000" "1"
+	ucidef_set_led_rssi "rssilow" "RSSILOW" "tp-link:green:link1" "wlan0" "1" "100"
+	ucidef_set_led_rssi "rssimediumlow" "RSSIMEDIUMLOW" "tp-link:green:link2" "wlan0" "30" "100"
+	ucidef_set_led_rssi "rssimediumhigh" "RSSIMEDIUMHIGH" "tp-link:green:link3" "wlan0" "60" "100"
+	ucidef_set_led_rssi "rssihigh" "RSSIHIGH" "tp-link:green:link4" "wlan0" "80" "100"
+	;;
 tplink,cpe210-v2|\
 tplink,cpe210-v3)
 	ucidef_set_led_netdev "lan" "LAN" "tp-link:green:lan" "eth0"
diff --git a/target/linux/ath79/base-files/etc/board.d/02_network b/target/linux/ath79/base-files/etc/board.d/02_network
index 33dc46b522..9c6429984f 100755
--- a/target/linux/ath79/base-files/etc/board.d/02_network
+++ b/target/linux/ath79/base-files/etc/board.d/02_network
@@ -102,6 +102,9 @@ ath79_setup_interfaces()
 		;;
 	comfast,cf-e110n-v2|\
 	comfast,cf-e120a-v3|\
+	tplink,cpe210-v1|\
+	tplink,cpe220-v2|\
+	tplink,cpe510-v1|\
 	ubnt,nanostation-m|\
 	ubnt,routerstation)
 		ucidef_set_interfaces_lan_wan "eth1" "eth0"
diff --git a/target/linux/ath79/base-files/etc/board.d/03_gpio_switches b/target/linux/ath79/base-files/etc/board.d/03_gpio_switches
index f588b29956..d84d6cfe85 100755
--- a/target/linux/ath79/base-files/etc/board.d/03_gpio_switches
+++ b/target/linux/ath79/base-files/etc/board.d/03_gpio_switches
@@ -42,6 +42,11 @@ tplink,archer-c25-v1)
 	ucidef_add_gpio_switch "led_control" "LED control" "21" "0"
 	ucidef_add_gpio_switch "led_reset" "LED reset" "19" "1"
 	;;
+tplink,cpe210-v1|\
+tplink,cpe220-v2|\
+tplink,cpe510-v1)
+	ucidef_add_gpio_switch "poe_passthrough" "PoE Passthrough" "20"
+	;;
 ubnt,nanostation-ac)
 	ucidef_add_gpio_switch "poe_passthrough" "PoE Passthrough" "3"
 	;;
diff --git a/target/linux/ath79/dts/ar9344_tplink_cpe210-v1.dts b/target/linux/ath79/dts/ar9344_tplink_cpe210-v1.dts
new file mode 100644
index 0000000000..1a107c23ef
--- /dev/null
+++ b/target/linux/ath79/dts/ar9344_tplink_cpe210-v1.dts
@@ -0,0 +1,9 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "ar9344_tplink_cpexxx-v1.dtsi"
+
+/ {
+	compatible = "tplink,cpe210-v1", "qca,ar9344";
+	model = "TP-Link CPE210 v1";
+};
diff --git a/target/linux/ath79/dts/ar9344_tplink_cpe220-v2.dts b/target/linux/ath79/dts/ar9344_tplink_cpe220-v2.dts
new file mode 100644
index 0000000000..0981b4af07
--- /dev/null
+++ b/target/linux/ath79/dts/ar9344_tplink_cpe220-v2.dts
@@ -0,0 +1,9 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "ar9344_tplink_cpexxx-v1.dtsi"
+
+/ {
+	compatible = "tplink,cpe220-v2", "qca,ar9344";
+	model = "TP-Link CPE220 v2";
+};
diff --git a/target/linux/ath79/dts/ar9344_tplink_cpe510-v1.dts b/target/linux/ath79/dts/ar9344_tplink_cpe510-v1.dts
new file mode 100644
index 0000000000..225d16e2e7
--- /dev/null
+++ b/target/linux/ath79/dts/ar9344_tplink_cpe510-v1.dts
@@ -0,0 +1,9 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "ar9344_tplink_cpexxx-v1.dtsi"
+
+/ {
+	compatible = "tplink,cpe510-v1", "qca,ar9344";
+	model = "TP-Link CPE510 v1";
+};
diff --git a/target/linux/ath79/dts/ar9344_tplink_cpe510.dtsi b/target/linux/ath79/dts/ar9344_tplink_cpe510.dtsi
index f19ef01d49..e1ffc29e83 100644
--- a/target/linux/ath79/dts/ar9344_tplink_cpe510.dtsi
+++ b/target/linux/ath79/dts/ar9344_tplink_cpe510.dtsi
@@ -1,10 +1,7 @@
 // SPDX-License-Identifier: GPL-2.0-or-later OR MIT
 /dts-v1/;
 
-#include <dt-bindings/gpio/gpio.h>
-#include <dt-bindings/input/input.h>
-
-#include "ar9344.dtsi"
+#include "ar9344_tplink_cpexxx.dtsi"
 
 / {
 	aliases {
@@ -42,99 +39,6 @@
 			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
 		};
 	};
-
-	keys {
-		compatible = "gpio-keys";
-
-		reset {
-			linux,code = <KEY_RESTART>;
-			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
-			debounce-interval = <60>;
-		};
-	};
-
-};
-
-&ref {
-	clock-frequency = <40000000>;
-};
-
-&uart {
-	status = "okay";
-};
-
-&spi {
-	num-cs = <1>;
-
-	status = "okay";
-
-	flash@0 {
-		compatible = "jedec,spi-nor";
-		reg = <0>;
-		spi-max-frequency = <25000000>;
-
-		partitions {
-			compatible = "fixed-partitions";
-			#address-cells = <1>;
-			#size-cells = <1>;
-
-			uboot: partition@0 {
-				label = "u-boot";
-				reg = <0x000000 0x020000>;
-				read-only;
-			};
-
-			partition@20000 {
-				label = "partition-table";
-				reg = <0x020000 0x010000>;
-				read-only;
-			};
-
-			info: partition@30000 {
-				label = "info";
-				reg = <0x030000 0x010000>;
-				read-only;
-			};
-
-			partition@40000 {
-				label = "firmware";
-				reg = <0x040000 0x780000>;
-
-				compatible = "fixed-partitions";
-				#address-cells = <1>;
-				#size-cells = <1>;
-
-				partition@0 {
-					label = "kernel";
-					reg = <0x000000 0x200000>;
-				};
-
-				partition@200000 {
-					label = "rootfs";
-					reg = <0x200000 0x580000>;
-				};
-			};
-
-			partition@7c0000 {
-				label = "config";
-				reg = <0x7c0000 0x030000>;
-				read-only;
-			};
-
-			art: partition@7f0000 {
-				label = "art";
-				reg = <0x7f0000 0x010000>;
-				read-only;
-			};
-		};
-	};
-};
-
-&wmac {
-	status = "okay";
-
-	mtd-cal-data = <&art 0x1000>;
-	mtd-mac-address = <&info 0x08>;
 };
 
 &eth0 {
diff --git a/target/linux/ath79/dts/ar9344_tplink_cpexxx-v1.dtsi b/target/linux/ath79/dts/ar9344_tplink_cpexxx-v1.dtsi
new file mode 100644
index 0000000000..81802f3b5b
--- /dev/null
+++ b/target/linux/ath79/dts/ar9344_tplink_cpexxx-v1.dtsi
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "ar9344_tplink_cpexxx.dtsi"
+
+/ {
+	aliases {
+		led-boot = &system;
+		led-failsafe = &system;
+		led-running = &system;
+		led-upgrade = &system;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		lan0 {
+			label = "tp-link:green:lan0";
+			gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
+		};
+
+		lan1 {
+			label = "tp-link:green:lan1";
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+		};
+
+		link1 {
+			label = "tp-link:green:link1";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+		};
+
+		link2 {
+			label = "tp-link:green:link2";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		};
+
+		link3 {
+			label = "tp-link:green:link3";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+
+		system: link4 {
+			label = "tp-link:green:link4";
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&gpio {
+	gpio_ext_lna0 {
+		gpio-hog;
+		gpios = <18 0>;
+		output-high;
+		line-name = "tp-link:ext:lna0";
+	};
+
+	gpio_ext_lna1 {
+		gpio-hog;
+		gpios = <19 0>;
+		output-high;
+		line-name = "tp-link:ext:lna1";
+	};
+};
+
+&eth1 {
+	status = "okay";
+
+	mtd-mac-address = <&info 0x8>;
+
+	gmac-config {
+		device = <&gmac>;
+		switch-phy-swap = <0>;
+		switch-only-mode = <1>;
+	};
+};
+
+&eth0 {
+	status = "okay";
+
+	phy-handle = <&swphy4>;
+
+	mtd-mac-address = <&info 0x8>;
+};
diff --git a/target/linux/ath79/dts/ar9344_tplink_cpexxx.dtsi b/target/linux/ath79/dts/ar9344_tplink_cpexxx.dtsi
new file mode 100644
index 0000000000..15b4fc8876
--- /dev/null
+++ b/target/linux/ath79/dts/ar9344_tplink_cpexxx.dtsi
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "ar9344.dtsi"
+
+/ {
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+};
+
+&ref {
+	clock-frequency = <40000000>;
+};
+
+&uart {
+	status = "okay";
+};
+
+&spi {
+	num-cs = <1>;
+
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			uboot: partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x020000>;
+				read-only;
+			};
+
+			partition@20000 {
+				label = "partition-table";
+				reg = <0x020000 0x010000>;
+				read-only;
+			};
+
+			info: partition@30000 {
+				label = "info";
+				reg = <0x030000 0x010000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "firmware";
+				reg = <0x040000 0x780000>;
+
+				compatible = "fixed-partitions";
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				partition@0 {
+					label = "kernel";
+					reg = <0x000000 0x200000>;
+				};
+
+				partition@200000 {
+					label = "rootfs";
+					reg = <0x200000 0x580000>;
+				};
+			};
+
+			partition@7c0000 {
+				label = "config";
+				reg = <0x7c0000 0x030000>;
+				read-only;
+			};
+
+			art: partition@7f0000 {
+				label = "art";
+				reg = <0x7f0000 0x010000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&wmac {
+	status = "okay";
+
+	mtd-cal-data = <&art 0x1000>;
+	mtd-mac-address = <&info 0x08>;
+};
diff --git a/target/linux/ath79/dts/qca9533_tplink_cpe210-v2.dts b/target/linux/ath79/dts/qca9533_tplink_cpe210-v2.dts
index e8273ba249..020dae67a7 100644
--- a/target/linux/ath79/dts/qca9533_tplink_cpe210-v2.dts
+++ b/target/linux/ath79/dts/qca9533_tplink_cpe210-v2.dts
@@ -5,5 +5,5 @@
 
 / {
 	compatible = "tplink,cpe210-v2", "qca,qca9533";
-	model = "TP-LINK CPE210 v2";
+	model = "TP-Link CPE210 v2";
 };
diff --git a/target/linux/ath79/dts/qca9533_tplink_cpe210-v3.dts b/target/linux/ath79/dts/qca9533_tplink_cpe210-v3.dts
index cf1f728d0f..fd24ecc2da 100644
--- a/target/linux/ath79/dts/qca9533_tplink_cpe210-v3.dts
+++ b/target/linux/ath79/dts/qca9533_tplink_cpe210-v3.dts
@@ -5,5 +5,5 @@
 
 / {
 	compatible = "tplink,cpe210-v3", "qca,qca9533";
-	model = "TP-LINK CPE210 v3";
+	model = "TP-Link CPE210 v3";
 };
diff --git a/target/linux/ath79/image/common-tp-link.mk b/target/linux/ath79/image/common-tp-link.mk
index da4616482a..d1d4a8940d 100644
--- a/target/linux/ath79/image/common-tp-link.mk
+++ b/target/linux/ath79/image/common-tp-link.mk
@@ -110,3 +110,12 @@ define Device/tplink-safeloader-uimage
   $(Device/tplink-safeloader)
   KERNEL := kernel-bin | append-dtb | lzma | uImageArcher lzma
 endef
+
+define Device/tplink-loader-okli
+  $(Device/tplink-safeloader)
+  LOADER_TYPE := elf
+  LOADER_FLASH_OFFS := 0x43000
+  COMPILE := loader-$(1).elf
+  COMPILE/loader-$(1).elf := loader-okli-compile
+  KERNEL := kernel-bin | append-dtb | lzma | uImage lzma -M 0x4f4b4c49 | loader-okli $(1) 12288
+endef
diff --git a/target/linux/ath79/image/generic-tp-link.mk b/target/linux/ath79/image/generic-tp-link.mk
index ba9f06adf6..71ea0acddf 100644
--- a/target/linux/ath79/image/generic-tp-link.mk
+++ b/target/linux/ath79/image/generic-tp-link.mk
@@ -157,6 +157,18 @@ define Device/tplink_archer-c7-v5
 endef
 TARGET_DEVICES += tplink_archer-c7-v5
 
+define Device/tplink_cpe210-v1
+  $(Device/tplink-loader-okli)
+  ATH_SOC := ar9344
+  IMAGE_SIZE := 7680k
+  DEVICE_MODEL := CPE210
+  DEVICE_VARIANT := v1
+  DEVICE_PACKAGES := rssileds
+  TPLINK_BOARD_ID := CPE210
+  SUPPORTED_DEVICES += cpe210
+endef
+TARGET_DEVICES += tplink_cpe210-v1
+
 define Device/tplink_cpe210-v2
   $(Device/tplink-safeloader)
   ATH_SOC := qca9533
@@ -183,52 +195,60 @@ define Device/tplink_cpe210-v3
 endef
 TARGET_DEVICES += tplink_cpe210-v3
 
+define Device/tplink_cpe220-v2
+  $(Device/tplink-loader-okli)
+  ATH_SOC := ar9344
+  IMAGE_SIZE := 7680k
+  DEVICE_MODEL := CPE220
+  DEVICE_VARIANT := v2
+  DEVICE_PACKAGES := rssileds
+  TPLINK_BOARD_ID := CPE220V2
+endef
+TARGET_DEVICES += tplink_cpe220-v2
+
+define Device/tplink_cpe510-v1
+  $(Device/tplink-loader-okli)
+  ATH_SOC := ar9344
+  IMAGE_SIZE := 7680k
+  DEVICE_MODEL := CPE510
+  DEVICE_VARIANT := v1
+  DEVICE_PACKAGES := rssileds
+  TPLINK_BOARD_ID := CPE510
+  SUPPORTED_DEVICES += cpe510
+endef
+TARGET_DEVICES += tplink_cpe510-v1
+
 define Device/tplink_cpe510-v2
-  $(Device/tplink-safeloader)
+  $(Device/tplink-loader-okli)
   ATH_SOC := ar9344
   IMAGE_SIZE := 7680k
   DEVICE_MODEL := CPE510
   DEVICE_VARIANT := v2
   DEVICE_PACKAGES := rssileds
   TPLINK_BOARD_ID := CPE510V2
-  LOADER_TYPE := elf
-  LOADER_FLASH_OFFS := 0x43000 
-  COMPILE := loader-$(1).elf 
-  COMPILE/loader-$(1).elf := loader-okli-compile 
-  KERNEL := kernel-bin | append-dtb | lzma | uImage lzma -M 0x4f4b4c49 | loader-okli $(1) 12288
   SUPPORTED_DEVICES += cpe510-v2
 endef
 TARGET_DEVICES += tplink_cpe510-v2
 
 define Device/tplink_cpe510-v3
-  $(Device/tplink-safeloader)
+  $(Device/tplink-loader-okli)
   ATH_SOC := ar9344
   IMAGE_SIZE := 7680k
   DEVICE_MODEL := CPE510
   DEVICE_VARIANT := v3
   DEVICE_PACKAGES := rssileds
   TPLINK_BOARD_ID := CPE510V3
-  LOADER_TYPE := elf
-  LOADER_FLASH_OFFS := 0x43000 
-  COMPILE := loader-$(1).elf 
-  COMPILE/loader-$(1).elf := loader-okli-compile 
-  KERNEL := kernel-bin | append-dtb | lzma | uImage lzma -M 0x4f4b4c49 | loader-okli $(1) 12288
   SUPPORTED_DEVICES += cpe510-v3
 endef
 TARGET_DEVICES += tplink_cpe510-v3
 
 define Device/tplink_cpe610-v1
-  $(Device/tplink-safeloader)
+  $(Device/tplink-loader-okli)
   ATH_SOC := ar9344
   IMAGE_SIZE := 7680k
   DEVICE_MODEL := CPE610
   DEVICE_VARIANT := v1
   TPLINK_BOARD_ID := CPE610V1
-  LOADER_TYPE := elf
-  LOADER_FLASH_OFFS := 0x43000
-  COMPILE := loader-$(1).elf
-  COMPILE/loader-$(1).elf := loader-okli-compile
-  KERNEL := kernel-bin | append-dtb | lzma | uImage lzma -M 0x4f4b4c49 | loader-okli $(1) 12288
 endef
 TARGET_DEVICES += tplink_cpe610-v1
 
diff --git a/tools/firmware-utils/src/tplink-safeloader.c b/tools/firmware-utils/src/tplink-safeloader.c
index 55f1ab9420..8dc72c9f43 100644
--- a/tools/firmware-utils/src/tplink-safeloader.c
+++ b/tools/firmware-utils/src/tplink-safeloader.c
@@ -118,7 +118,7 @@ static const uint8_t md5_salt[16] = {
 
 /** Firmware layout table */
 static struct device_info boards[] = {
-	/** Firmware layout for the CPE210/220 */
+	/** Firmware layout for the CPE210/220 V1 */
 	{
 		.id	= "CPE210",
 		.vendor	= "CPE510(TP-LINK|UN|N300-5):1.0\r\n",
@@ -231,6 +231,46 @@ static struct device_info boards[] = {
 		.last_sysupgrade_partition = "support-list",
 	},
 
+	/** Firmware layout for the CPE220 V2 */
+	{
+		.id	= "CPE220V2",
+		.vendor	= "CPE510(TP-LINK|UN|N300-5):1.0\r\n",
+		.support_list =
+			"SupportList:\r\n"
+			"CPE220(TP-LINK|EU|N300-2|00000000):2.0\r\n"
+			"CPE220(TP-LINK|EU|N300-2|45550000):2.0\r\n"
+			"CPE220(TP-LINK|EU|N300-2|55530000):2.0\r\n"
+			"CPE220(TP-LINK|UN|N300-2|00000000):2.0\r\n"
+			"CPE220(TP-LINK|UN|N300-2|45550000):2.0\r\n"
+			"CPE220(TP-LINK|UN|N300-2|55530000):2.0\r\n"
+			"CPE220(TP-LINK|US|N300-2|55530000):2.0\r\n"
+			"CPE220(TP-LINK|UN|N300-2):2.0\r\n"
+			"CPE220(TP-LINK|EU|N300-2):2.0\r\n"
+			"CPE220(TP-LINK|US|N300-2):2.0\r\n",
+		.support_trail = '\xff',
+		.soft_ver = NULL,
+
+		.partitions = {
+			{"fs-uboot", 0x00000, 0x20000},
+			{"partition-table", 0x20000, 0x02000},
+			{"default-mac", 0x30000, 0x00020},
+			{"product-info", 0x31100, 0x00100},
+			{"signature", 0x32000, 0x00400},
+			{"os-image", 0x40000, 0x200000},
+			{"file-system", 0x240000, 0x570000},
+			{"soft-version", 0x7b0000, 0x00100},
+			{"support-list", 0x7b1000, 0x00400},
+			{"user-config", 0x7c0000, 0x10000},
+			{"default-config", 0x7d0000, 0x10000},
+			{"log", 0x7e0000, 0x10000},
+			{"radio", 0x7f0000, 0x10000},
+			{NULL, 0, 0}
+		},
+
+		.first_sysupgrade_partition = "os-image",
+		.last_sysupgrade_partition = "support-list",
+	},
+
 	/** Firmware layout for the CPE510/520 */
 	{
 		.id	= "CPE510",
