From: template <nomail@local>
Date: Fri, 6 Sep 2019 08:29:50 +0000
Subject: imagebuilder-custom-postinst-script

diff --git a/target/imagebuilder/files/Makefile b/target/imagebuilder/files/Makefile
index 835dd98ca5fe9f876247d3bf43c02459e640dcd1..59f99d13ab32634ef29591743d1d3b4dc020ee50 100644
--- a/target/imagebuilder/files/Makefile
+++ b/target/imagebuilder/files/Makefile
@@ -117,6 +117,9 @@ _call_image: staging_dir/host/.prereq-build
 	$(MAKE) package_reload
 	$(MAKE) package_install
 	$(MAKE) -s prepare_rootfs
+ifneq ($(CUSTOM_POSTINST_SCRIPT),)
+	(cd "$(TARGET_DIR)" ; bash $(CUSTOM_POSTINST_SCRIPT))
+endif
 	$(MAKE) -s build_image
 	$(MAKE) -s json_overview_image_info
 	$(MAKE) -s checksum
