diff --git a/target/imagebuilder/files/Makefile b/target/imagebuilder/files/Makefile
index 95df8bf113..0debd5a882 100644
--- a/target/imagebuilder/files/Makefile
+++ b/target/imagebuilder/files/Makefile
@@ -105,6 +105,9 @@ _call_image: staging_dir/host/.prereq-build
 	$(MAKE) package_reload
 	$(MAKE) package_install
 	$(MAKE) -s prepare_rootfs
+ifneq ($(CUSTOM_POSTINST_SCRIPT),)
+	(cd "$(TARGET_DIR)" ; bash $(CUSTOM_POSTINST_SCRIPT))
+endif
 	$(MAKE) -s build_image
 	$(MAKE) -s checksum
 
