--- lede/package/base-files/files/etc/banner	2017-12-10 21:27:21.352612551 +0100
+++ /tmp/banner	2017-12-10 21:19:20.000000000 +0100
@@ -1,8 +1,20 @@
-  _______                     ________        __
- |       |.-----.-----.-----.|  |  |  |.----.|  |_
- |   -   ||  _  |  -__|     ||  |  |  ||   _||   _|
- |_______||   __|_____|__|__||________||__|  |____|
-          |__| W I R E L E S S   F R E E D O M
- -----------------------------------------------------
- %D %V, %C
+  _____        _  __             _
+ |  ___|      (_)/ _|           | |
+ | |_ _ __ ___ _| |_ _   _ _ __ | | __
+ |  _| '__/ _ \ |  _| | | | '_ \| |/ /
+ | | | | |  __/ | | | |_| | | | |   <
+ \_| |_|  \___|_|_|  \__,_|_| |_|_|\_\
+
+ Firmware %D (%V rev %C)
+ %P - %S
+ https://wiki.freifunk.net/Berlin:Firmware
+ https://github.com/freifunk-berlin
  -----------------------------------------------------
+
+ If you find bugs please report them at:
+
+   https://github.com/freifunk-berlin/firmware/issues
+
+ For questions write a mail to <berlin@berlin.freifunk.net>
+ or check https://berlin.freifunk.net/contact for our weekly meetings.
+
