diff -urN fritz.pci.orig/src/driver.c fritz.pci/src/driver.c
--- fritz.pci.orig/src/driver.c	2007-09-15 13:48:19.000000000 +0200
+++ fritz.pci/src/driver.c	2007-09-15 13:50:49.000000000 +0200
@@ -107,7 +107,7 @@
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
 static void scheduler (unsigned long data);
-static irqreturn_t irq_handler (int irq, void * args, struct pt_regs * regs);
+static irqreturn_t irq_handler (int irq, void * args);
 
 static DECLARE_TASKLET_DISABLED (scheduler_tasklet, scheduler, 0);
 
@@ -836,11 +836,10 @@
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-static irqreturn_t irq_handler (int irq, void * args, struct pt_regs * regs) {
+static irqreturn_t irq_handler (int irq, void * args) {
 	int	res	= IRQ_NONE;
 	
 	UNUSED_ARG (irq);
-	UNUSED_ARG (regs);
 	if (args != NULL) {
 		assert (capi_lib->cm_handle_events != NULL);
 		if (atomic_read (&scheduler_id) == smp_processor_id ()) {
@@ -901,14 +900,14 @@
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-int driver_init (void) {
+int fritz_driver_init (void) {
 
 	return (NULL != (capi_lib = link_library (NULL)));
 } /* driver_init */
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-void driver_exit (void) {
+void fritz_driver_exit (void) {
 
 	assert (capi_lib != NULL);
 	free_library ();
diff -urN fritz.pci.orig/src/driver.h fritz.pci/src/driver.h
--- fritz.pci.orig/src/driver.h	2007-09-15 13:48:19.000000000 +0200
+++ fritz.pci/src/driver.h	2007-09-15 13:49:41.000000000 +0200
@@ -23,7 +23,6 @@
 #ifndef __have_driver_h__
 #define __have_driver_h__
 
-#include <linux/config.h>
 #include <linux/skbuff.h>
 #include <linux/capi.h>
 #include <linux/list.h>
@@ -98,8 +97,8 @@
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
-extern int driver_init (void);
-extern void driver_exit (void);
+extern int fritz_driver_init (void);
+extern void fritz_driver_exit (void);
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
diff -urN fritz.pci.orig/src/main.c fritz.pci/src/main.c
--- fritz.pci.orig/src/main.c	2007-09-15 13:48:19.000000000 +0200
+++ fritz.pci/src/main.c	2007-09-15 13:49:41.000000000 +0200
@@ -22,7 +22,6 @@
 
 #include <stdarg.h>
 #include <asm/uaccess.h>
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
@@ -205,7 +204,7 @@
 	NOTE(PRODUCT_LOGO " found: port 0x%04x, irq %u\n", pars.port, pars.irq);
 
 	NOTE("Loading...\n");
-	if (!driver_init ()) {
+	if (!fritz_driver_init ()) {
 		ERROR("Error: Driver library not available.\n");
 		ERROR("Not loaded.\n");
 		return res;
@@ -238,7 +237,7 @@
 	remove_ctrl (&card->ctrl);
 	NOTE("Removed.\n");
 	libheap_exit ();
-	driver_exit ();
+	fritz_driver_exit ();
 #ifndef NDEBUG
 	if (hallocated() != 0) {
 		ERROR("%u bytes leaked.\n", hallocated());
@@ -270,7 +269,7 @@
 	NOTE (PRODUCT_LOGO " found: port 0x%04x, irq %u\n", pars.port, pars.irq);
 
 	NOTE("Loading...\n");
-	if (!driver_init ()) {
+	if (!fritz_driver_init ()) {
 		ERROR("Error: Driver library not available.\n");
 		ERROR("Not loaded.\n");
 		return -EBUSY;
@@ -303,7 +302,7 @@
 	remove_ctrl (&card->ctrl);
 	NOTE("Removed.\n");
 	libheap_exit ();
-	driver_exit ();
+	fritz_driver_exit ();
 #ifndef NDEBUG
 	if (hallocated() != 0) {
 		ERROR("%u bytes leaked.\n", hallocated());
@@ -364,7 +363,7 @@
 #endif
 
 #if defined (__fcpci__)	
-	if (0 == (err = pci_module_init (&fcpci_driver))) {
+	if (0 == (err = pci_register_driver (&fcpci_driver))) {
 		LOG("PCI driver registered.\n");
 		register_capi_driver (&fritz_capi_driver);
 		LOG("CAPI driver registered.\n");
@@ -378,7 +377,7 @@
 	}
 	return err;
 #elif defined (__fcpcmcia__) || defined (__fcclassic__)
-	if (!driver_init ()) {
+	if (!fritz_driver_init ()) {
 		ERROR("Error: Driver library not available.\n");
 		ERROR("Not loaded.\n");
 		return -EBUSY;
diff -urN fritz.pci.orig/src/Makefile fritz.pci/src/Makefile
--- fritz.pci.orig/src/Makefile	2007-09-15 13:48:19.000000000 +0200
+++ fritz.pci/src/Makefile	2007-09-15 13:49:41.000000000 +0200
@@ -1,14 +1,14 @@
 # COMMON
 
 CARD		:= fcpci
-LIBDIR		:= /var/lib/fritz
+LIBDIR		?= /var/lib/fritz
 OBJECTS		:= main.o driver.o tools.o tables.o queue.o lib.o
 
 ifneq ($(KERNELRELEASE),)
 
 # KERNEL
 
-EXTRA_CFLAGS	+= -D__$(CARD)__ -DTARGET=\"$(CARD)\"
+EXTRA_CFLAGS	+= -D__$(CARD)__ -DTARGET=\"$(CARD)\" @CFLAGS@
 ifndef DEBUG
 EXTRA_CFLAGS	+= -DNDEBUG
 endif
diff -urN fritz.pci.orig/src/tools.h fritz.pci/src/tools.h
--- fritz.pci.orig/src/tools.h	2007-09-15 13:48:19.000000000 +0200
+++ fritz.pci/src/tools.h	2007-09-15 13:49:41.000000000 +0200
@@ -71,12 +71,14 @@
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
+#ifndef atomic_xchg
 static inline unsigned long atomic_xchg (
 	volatile atomic_t *	v, 
 	unsigned		value
 ) {
 	return __xchg (value, &v->counter, sizeof (unsigned));
 } /* atomic_xchg */
+#endif
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
--- fli4l/src/kernel-2.6.2/build-2.6.24/fcpci-3.11-07/fritz/src/driver.c	2008-02-04 17:04:11.000000000 +0100
+++ fritz.pci/src/driver.c	2008-02-04 17:11:13.000000000 +0100
@@ -194,7 +194,7 @@
 	}
 	lib_strncpy (ctrl->serial, tmp, CAPI_SERIAL_LEN);
 	lib_memcpy (&ctrl->profile, card->string[6], sizeof (capi_profile));
-	strncpy (ctrl->manu, "AVM GmbH", CAPI_MANUFACTURER_LEN);
+	lib_strncpy (ctrl->manu, "AVM GmbH", CAPI_MANUFACTURER_LEN);
 	ctrl->version.majorversion = 2;
 	ctrl->version.minorversion = 0;
 	tmp = card->string[0];
@@ -374,9 +374,9 @@
 			card->irq, 
 			&irq_handler, 
 #if defined (__fcpci__) || defined (__fcpcmcia__)
-			SA_INTERRUPT | SA_SHIRQ, 
+			IRQF_DISABLED | IRQF_SHARED, 
 #else
-			SA_INTERRUPT, 
+			IRQF_DISABLED, 
 #endif
 			TARGET, 
 			card
