From: Sven Roederer <devel-sven@geroedel.de>
Date: Sat, 24 Apr 2021 02:03:40 +0200
Subject: uplink/notunnel: fix syntax error in package description text

diff --git a/uplinks/freifunk-berlin-uplink-notunnel/Makefile b/uplinks/freifunk-berlin-uplink-notunnel/Makefile
index ec9c2c222e77634471197d135faab110ffbf851d..3bf2896d10a210057641308ef3e21eabb91a9074 100644
--- a/uplinks/freifunk-berlin-uplink-notunnel/Makefile
+++ b/uplinks/freifunk-berlin-uplink-notunnel/Makefile
@@ -4,8 +4,7 @@ LUCI_TITLE:=Freifunk Berlin no tunnel files
 LUCI_DEPENDS:=+freifunk-berlin-lib-guard +kmod-veth +freifunk-berlin-network-defaults +pingcheck
 PKG_PROVIDES:=freifunk-berlin-uplink
 
-LUCI_DESCRIPTION:=Freifunk Berlin files to setup a uplink without any
-  tunnel.
+LUCI_DESCRIPTION:=Freifunk Berlin files to setup a uplink without any tunnel.\
   upstream traffic will be directly routed via your network
 
 include ../../freifunk-berlin-generic.mk
