diff --git a/defaults/freifunk-berlin-dhcp-defaults/Makefile b/defaults/freifunk-berlin-dhcp-defaults/Makefile
index 6e5daf788..f1232b1a2 100644
--- a/defaults/freifunk-berlin-dhcp-defaults/Makefile
+++ b/defaults/freifunk-berlin-dhcp-defaults/Makefile
@@ -1,12 +1,8 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=freifunk-berlin-dhcp-defaults
-PKG_VERSION:=0.0.3
-PKG_RELEASE:=1
 
-PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)
-
-include $(INCLUDE_DIR)/package.mk
+include ../../freifunk-berlin-generic.mk
 
 define Package/freifunk-berlin-dhcp-defaults
   SECTION:=freifunk-berlin
diff --git a/defaults/freifunk-berlin-firewall-defaults/Makefile b/defaults/freifunk-berlin-firewall-defaults/Makefile
index 36d176896..3f2ec54cd 100644
--- a/defaults/freifunk-berlin-firewall-defaults/Makefile
+++ b/defaults/freifunk-berlin-firewall-defaults/Makefile
@@ -1,12 +1,8 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=freifunk-berlin-firewall-defaults
-PKG_VERSION:=0.0.6
-PKG_RELEASE:=1
 
-PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)
-
-include $(INCLUDE_DIR)/package.mk
+include ../../freifunk-berlin-generic.mk
 
 define Package/freifunk-berlin-firewall-defaults
   SECTION:=freifunk-berlin
diff --git a/defaults/freifunk-berlin-freifunk-defaults/Makefile b/defaults/freifunk-berlin-freifunk-defaults/Makefile
index 504cb50ca..3fcccfa41 100644
--- a/defaults/freifunk-berlin-freifunk-defaults/Makefile
+++ b/defaults/freifunk-berlin-freifunk-defaults/Makefile
@@ -1,12 +1,8 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=freifunk-berlin-freifunk-defaults
-PKG_VERSION:=0.0.2
-PKG_RELEASE:=3
 
-PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)
-
-include $(INCLUDE_DIR)/package.mk
+include ../../freifunk-berlin-generic.mk
 
 define Package/freifunk-berlin-freifunk-defaults
   SECTION:=freifunk-berlin
diff --git a/defaults/freifunk-berlin-network-defaults/Makefile b/defaults/freifunk-berlin-network-defaults/Makefile
index ea6427df7..d26c2e8ff 100644
--- a/defaults/freifunk-berlin-network-defaults/Makefile
+++ b/defaults/freifunk-berlin-network-defaults/Makefile
@@ -1,12 +1,8 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=freifunk-berlin-network-defaults
-PKG_VERSION:=0.2.0
-PKG_RELEASE:=1
 
-PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)
-
-include $(INCLUDE_DIR)/package.mk
+include ../../freifunk-berlin-generic.mk
 
 define Package/freifunk-berlin-network-defaults
   SECTION:=freifunk-berlin
diff --git a/defaults/freifunk-berlin-olsrd-defaults/Makefile b/defaults/freifunk-berlin-olsrd-defaults/Makefile
index 269bbb7de..87f442e58 100644
--- a/defaults/freifunk-berlin-olsrd-defaults/Makefile
+++ b/defaults/freifunk-berlin-olsrd-defaults/Makefile
@@ -1,12 +1,8 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=freifunk-berlin-olsrd-defaults
-PKG_VERSION:=0.0.6
-PKG_RELEASE:=2
 
-PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)
-
-include $(INCLUDE_DIR)/package.mk
+include ../../freifunk-berlin-generic.mk
 
 define Package/freifunk-berlin-olsrd-defaults
   SECTION:=freifunk-berlin
diff --git a/defaults/freifunk-berlin-openvpn-files/Makefile b/defaults/freifunk-berlin-openvpn-files/Makefile
index 881cdd1f8..8c8f4cc5a 100644
--- a/defaults/freifunk-berlin-openvpn-files/Makefile
+++ b/defaults/freifunk-berlin-openvpn-files/Makefile
@@ -3,12 +3,8 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=freifunk-berlin-openvpn-files
-PKG_VERSION:=0.1.1
-PKG_RELEASE:=1
 
-PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)
-
-include $(INCLUDE_DIR)/package.mk
+include ../../freifunk-berlin-generic.mk
 
 define Package/freifunk-berlin-openvpn-files
   SECTION:=freifunk-berlin
diff --git a/defaults/freifunk-berlin-statistics-defaults/Makefile b/defaults/freifunk-berlin-statistics-defaults/Makefile
index 974b554ac..831bfb110 100644
--- a/defaults/freifunk-berlin-statistics-defaults/Makefile
+++ b/defaults/freifunk-berlin-statistics-defaults/Makefile
@@ -1,12 +1,8 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=freifunk-berlin-statistics-defaults
-PKG_VERSION:=0.0.2
-PKG_RELEASE:=1
 
-PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)
-
-include $(INCLUDE_DIR)/package.mk
+include ../../freifunk-berlin-generic.mk
 
 define Package/freifunk-berlin-statistics-defaults
   SECTION:=freifunk-berlin
diff --git a/defaults/freifunk-berlin-system-defaults/Makefile b/defaults/freifunk-berlin-system-defaults/Makefile
index 9fcfe007f..adeb567a5 100644
--- a/defaults/freifunk-berlin-system-defaults/Makefile
+++ b/defaults/freifunk-berlin-system-defaults/Makefile
@@ -1,12 +1,8 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=freifunk-berlin-system-defaults
-PKG_VERSION:=0.0.2
-PKG_RELEASE:=1
 
-PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)
-
-include $(INCLUDE_DIR)/package.mk
+include ../../freifunk-berlin-generic.mk
 
 define Package/freifunk-berlin-system-defaults
   SECTION:=freifunk-berlin
diff --git a/defaults/freifunk-berlin-uhttpd-defaults/Makefile b/defaults/freifunk-berlin-uhttpd-defaults/Makefile
index 32a75b115..df98843b1 100644
--- a/defaults/freifunk-berlin-uhttpd-defaults/Makefile
+++ b/defaults/freifunk-berlin-uhttpd-defaults/Makefile
@@ -1,12 +1,8 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=freifunk-berlin-uhttpd-defaults
-PKG_VERSION:=0.0.1
-PKG_RELEASE:=2
 
-PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)
-
-include $(INCLUDE_DIR)/package.mk
+include ../../freifunk-berlin-generic.mk
 
 define Package/freifunk-berlin-uhttpd-defaults
   SECTION:=freifunk-berlin
diff --git a/freifunk-berlin-generic.mk b/freifunk-berlin-generic.mk
new file mode 100644
index 000000000..82d5b6025
--- /dev/null
+++ b/freifunk-berlin-generic.mk
@@ -0,0 +1,35 @@
+#
+# This is free software, licensed under the GNU General Public License v3.0 .
+#
+
+LUCIMKFILE:=$(wildcard $(TOPDIR)/feeds/*/luci.mk)
+
+# verify that there is only one single file returned
+ifneq (1,$(words $(LUCIMKFILE)))
+ifeq (0,$(words $(LUCIMKFILE)))
+$(error did not find luci.mk in any feed)
+else
+$(error found multiple luci.mk files in the feeds)
+endif
+else
+#$(info found luci.mk at $(LUCIMKFILE))
+endif
+
+PKG_VERSION?=$(if $(DUMP),x,$(strip $(shell \
+	if svn info >/dev/null 2>/dev/null; then \
+		revision="svn-r$$(LC_ALL=C svn info | sed -ne 's/^Revision: //p')"; \
+	elif git log -1 >/dev/null 2>/dev/null; then \
+		revision="svn-r$$(LC_ALL=C git log -1 | sed -ne 's/.*git-svn-id: .*@\([0-9]\+\) .*/\1/p')"; \
+		if [ "$$revision" = "svn-r" ]; then \
+			set -- $$(git log -1 --format="%ct %h" --abbrev=7 .); \
+			secs="$$(($$1 % 86400))"; \
+			yday="$$(date --utc --date="@$$1" "+%y.%j")"; \
+			revision="$$(printf 'git-%s.%05d-%s' "$$yday" "$$secs" "$$2")"; \
+		fi; \
+	else \
+		revision="unknown"; \
+	fi; \
+	echo "$$revision" \
+)))
+
+include $(LUCIMKFILE)
