diff --git a/utils/luci-app-ffwizard-berlin/Makefile b/utils/luci-app-ffwizard-berlin/Makefile
index fa1d867b7..febadfd00 100644
--- a/utils/luci-app-ffwizard-berlin/Makefile
+++ b/utils/luci-app-ffwizard-berlin/Makefile
@@ -2,55 +2,12 @@
 
 include $(TOPDIR)/rules.mk
 
-PKG_NAME:=luci-app-ffwizard-berlin
-PKG_VERSION:=0.0.16
-PKG_RELEASE:=1
+LUCI_TITLE:=Freifunk Berlin configuration wizard
+LUCI_DEPENDS:=+luci-mod-admin-full +freifunk-policyrouting +luci-lib-jsonc +luci-lib-ipkg
+PKG_RELEASE:=4
 
-PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)
+include ../../freifunk-berlin-generic.mk
 
-include $(INCLUDE_DIR)/package.mk
+#  URL:=http://berlin.freifunk.net
 
-define Package/luci-app-ffwizard-berlin
-  SECTION:=luci
-  CATEGORY:=LuCI
-  SUBMENU:=3. Applications
-  TITLE:=Freifunk Berlin configuration wizard
-  URL:=http://berlin.freifunk.net
-  DEPENDS+= +luci-mod-admin-full +freifunk-policyrouting +luci-lib-jsonc +luci-lib-ipkg
-  PKGARCH:=all
-endef
-
-define Package/luci-app-ffwizard-berlin/description
-  Freifunk Wizard for Berlin
-endef
-
-define Build/Prepare
-	mkdir -p $(PKG_BUILD_DIR)
-	$(CP) ./luasrc $(PKG_BUILD_DIR)/
-endef
-
-define Build/Configure
-endef
-
-define Build/Compile
-	$(MAKE) -C $(PKG_BUILD_DIR)/luasrc
-endef
-
-define Package/luci-app-ffwizard-berlin/install
-	$(INSTALL_DIR) $(1)/usr/lib/lua/luci/controller
-	$(INSTALL_DIR) $(1)/usr/lib/lua/luci/model
-	$(INSTALL_DIR) $(1)/usr/lib/lua/luci/tools
-	$(INSTALL_DIR) $(1)/usr/lib/lua/luci/view
-	$(CP) $(PKG_BUILD_DIR)/luasrc/controller/* $(1)/usr/lib/lua/luci/controller
-	$(CP) $(PKG_BUILD_DIR)/luasrc/model/* $(1)/usr/lib/lua/luci/model
-	$(CP) $(PKG_BUILD_DIR)/luasrc/tools/* $(1)/usr/lib/lua/luci/tools
-	$(CP) $(PKG_BUILD_DIR)/luasrc/view/* $(1)/usr/lib/lua/luci/view
-	$(INSTALL_DIR) $(1)/usr/share/rpcd/acl.d/
-	$(CP) ./root/usr/share/rpcd/acl.d/ffwizard-berlin.json $(1)/usr/share/rpcd/acl.d/ffwizard-berlin.json
-	$(INSTALL_DIR) $(1)/etc
-	$(CP) ./root/etc/* $(1)/etc
-	$(INSTALL_DIR) $(1)/usr/libexec/rpcd
-	$(CP) ./root/usr/libexec/rpcd/* $(1)/usr/libexec/rpcd/
-endef
-
-$(eval $(call BuildPackage,luci-app-ffwizard-berlin))
+# call BuildPackage - OpenWrt buildroot signature
diff --git a/utils/luci-mod-freifunk-ui/Makefile b/utils/luci-mod-freifunk-ui/Makefile
index e943501e8..21aa211cb 100644
--- a/utils/luci-mod-freifunk-ui/Makefile
+++ b/utils/luci-mod-freifunk-ui/Makefile
@@ -7,44 +7,10 @@
 
 include $(TOPDIR)/rules.mk
 
-PKG_NAME:=luci-mod-freifunk-ui
-PKG_VERSION:=0.0.4
-PKG_RELEASE:=1
+LUCI_TITLE:=Freifunk Public and Admin LuCI UI
+LUCI_DEPENDS:=+luci-mod-admin-full +luci-lib-json +community-profiles +luci-lib-ipkg
+LUCI_PKGARCH:=all
 
-PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)
+include ../../freifunk-berlin-generic.mk
 
-include $(INCLUDE_DIR)/package.mk
-
-define Package/luci-mod-freifunk-ui
-  SECTION:=luci
-  CATEGORY:=LuCI
-  SUBMENU:=2. Modules
-  TITLE:=Freifunk Public and Admin LuCI UI
-  DEPENDS:=+luci-mod-admin-full +luci-lib-json +community-profiles +luci-lib-ipkg
-  PKGARCH:=all
-endef
-
-define Build/Prepare
-	mkdir -p $(PKG_BUILD_DIR)
-	$(CP) ./luasrc $(PKG_BUILD_DIR)/
-endef
-
-define Build/Configure
-endef
-
-define Build/Compile
-#	$(MAKE) -C $(PKG_BUILD_DIR)/luasrc
-endef
-
-define Package/luci-mod-freifunk-ui/install
-	$(INSTALL_DIR) $(1)/usr/lib/lua/luci/controller
-	$(INSTALL_DIR) $(1)/usr/lib/lua/luci/model
-	$(INSTALL_DIR) $(1)/usr/lib/lua/luci/view
-	$(CP) $(PKG_BUILD_DIR)/luasrc/controller/* $(1)/usr/lib/lua/luci/controller
-	$(CP) $(PKG_BUILD_DIR)/luasrc/model/* $(1)/usr/lib/lua/luci/model
-	$(CP) $(PKG_BUILD_DIR)/luasrc/view/* $(1)/usr/lib/lua/luci/view
-	$(INSTALL_DIR) $(1)/www
-	$(CP) ./htdocs/* $(1)/www
-endef
-
-$(eval $(call BuildPackage,luci-mod-freifunk-ui))
+# call BuildPackage - OpenWrt buildroot signature
