Index: src/uhttpd.c
===================================================================
--- src/uhttpd.c	(Revision 21119)
+++ src/uhttpd.c	(Arbeitskopie)
@@ -912,8 +912,14 @@
 						/* 404 */
 						else
 						{
-							uh_http_sendhf(cl, 404, "Not Found",
-								"No such file or directory");
+							//splash
+							if( (pin = uh_path_lookup(cl, "/luci/splash/index.html")) != NULL )
+							{
+								if( uh_auth_check(cl, req, pin) )
+								{
+									uh_file_request(cl, req, pin);
+								}
+							}
 						}
 					}
 
Index: Makefile
===================================================================
--- Makefile	(Revision 21119)
+++ Makefile	(Arbeitskopie)
@@ -8,7 +8,7 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=uhttpd
-PKG_RELEASE:=8
+PKG_RELEASE:=9
 
 PKG_BUILD_DIR := $(BUILD_DIR)/$(PKG_NAME)
 
